{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "24fb07d6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "int32\n",
      "(6, 6)\n",
      "0\n",
      "1\n",
      "[[0 0 1 0]\n",
      " [1 0 1 0]\n",
      " [0 1 1 0]\n",
      " [0 0 0 0]]\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "Z = np.array([[0,0,0,0,0,0],\n",
    "                  [0,0,0,1,0,0],\n",
    "                  [0,1,0,1,0,0],\n",
    "                  [0,0,1,1,0,0],\n",
    "                  [0,0,0,0,0,0],\n",
    "                  [0,0,0,0,0,1]])\n",
    "print(Z.dtype)\n",
    "print(Z.shape)\n",
    "print(Z[0,5])\n",
    "print(Z[1,3])\n",
    "print(Z[1:5,1:5])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ead25f0d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[0 0 0]\n",
      " [0 0 0]]\n"
     ]
    }
   ],
   "source": [
    "a = np.array([[0, 0, 0], [0, 0, 0]], dtype=int)\n",
    "print(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7124a619",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.1.5'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7e9571e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ 4,  7, -5,  3], dtype=int64)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "my_obj = pd.Series([4, 7, -5, 3])\n",
    "my_obj.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "8966f763",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "RangeIndex(start=0, stop=4, step=1)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "my_obj.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1a5b828d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "a     8\n",
       "b     9\n",
       "c    10\n",
       "d    11\n",
       "dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "my_obj2 = pd.Series([8,9,10,11], index=['a','b','c','d'])\n",
    "my_obj2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "577a27a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'a' in my_obj2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "d1f1bf0c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "name              apple\n",
       "birthday       1996-1-1\n",
       "luckynumber           7\n",
       "dtype: object"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dic_data = {'name':'apple','birthday':'1996-1-1','luckynumber':7 }\n",
    "my_obj3 = pd.Series(dic_data)\n",
    "my_obj3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c30ae5bb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is a test"
     ]
    }
   ],
   "source": [
    "file =open('test.txt','r',encoding='UTF-8')\n",
    "try:\n",
    "    for text in file:\n",
    "        print(text,end='')\n",
    "except:\n",
    "    print(\"讀檔失敗\")\n",
    "finally:\n",
    "    file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b561c90f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is a test"
     ]
    }
   ],
   "source": [
    "with open('test.txt','r',encoding='UTF-8') as file:\n",
    "    for text in file:\n",
    "        print(text,end='')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "1a509416",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     True\n",
       "1    False\n",
       "2     True\n",
       "3     True\n",
       "dtype: bool"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "registration = [True,False,True,True]\n",
    "registration = pd.Series(registration)\n",
    "registration"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "1f3d75f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>year</th>\n",
       "      <th>month</th>\n",
       "      <th>day</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Bob</td>\n",
       "      <td>1996</td>\n",
       "      <td>8</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Nancy</td>\n",
       "      <td>1997</td>\n",
       "      <td>8</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Amy</td>\n",
       "      <td>1997</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Elsa</td>\n",
       "      <td>1996</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Jack</td>\n",
       "      <td>1997</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    name  year  month  day\n",
       "0    Bob  1996      8   11\n",
       "1  Nancy  1997      8   23\n",
       "2    Amy  1997      7    8\n",
       "3   Elsa  1996      1    3\n",
       "4   Jack  1997     12   11"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = {'name': ['Bob', 'Nancy','Amy','Elsa','Jack'],\n",
    "        'year': [1996, 1997, 1997, 1996, 1997],\n",
    "        'month': [8, 8, 7, 1, 12],\n",
    "        'day':[11,23,8,3,11]}\n",
    "myframe = pd.DataFrame(data)\n",
    "myframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "05383bdb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-21-f64154864bb0>\", line 1, in <module>\n",
      "    myframe2 = pd.DataFrame(data,columns=['name','year', 'day', 'month'])\n",
      "NameError: name 'data' is not defined\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'NameError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "ERROR! Session/line number was not unique in database. History logging moved to new session 84\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-21-f64154864bb0>\", line 1, in <module>\n",
      "    myframe2 = pd.DataFrame(data,columns=['name','year', 'day', 'month'])\n",
      "NameError: name 'data' is not defined\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'NameError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3263, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3360, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1193, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-21-f64154864bb0>\", line 1, in <module>\n",
      "    myframe2 = pd.DataFrame(data,columns=['name','year', 'day', 'month'])\n",
      "NameError: name 'data' is not defined\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'NameError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3263, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3360, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1193, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2895, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3072, in run_cell_async\n",
      "    interactivity=interactivity, compiler=compiler, result=result)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3282, in run_ast_nodes\n",
      "    self.showtraceback()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1211, in structured_traceback\n",
      "    chained_exceptions_tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n"
     ]
    }
   ],
   "source": [
    "myframe2 = pd.DataFrame(data,columns=['name','year', 'day', 'month'])\n",
    "myframe2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "257e02dc",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-21-47deb6e12694>\", line 1, in <module>\n",
      "    myframe2[['name','year']][1:3]\n",
      "NameError: name 'myframe2' is not defined\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'NameError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "ERROR! Session/line number was not unique in database. History logging moved to new session 85\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-21-47deb6e12694>\", line 1, in <module>\n",
      "    myframe2[['name','year']][1:3]\n",
      "NameError: name 'myframe2' is not defined\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'NameError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3263, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3360, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1193, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-21-47deb6e12694>\", line 1, in <module>\n",
      "    myframe2[['name','year']][1:3]\n",
      "NameError: name 'myframe2' is not defined\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'NameError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3263, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3360, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1193, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2895, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3072, in run_cell_async\n",
      "    interactivity=interactivity, compiler=compiler, result=result)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3282, in run_ast_nodes\n",
      "    self.showtraceback()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1211, in structured_traceback\n",
      "    chained_exceptions_tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n"
     ]
    }
   ],
   "source": [
    "myframe2[['name','year']][1:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "bf830e05",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'myframe2' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-19-a4bfc37c8f71>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[0mluckynumber\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;34m'3'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'2'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'1'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'7'\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;34m'8'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[0mluckynumber\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mSeries\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mluckynumber\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 3\u001b[1;33m \u001b[0mmyframe2\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'luckynumber'\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mluckynumber\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      4\u001b[0m \u001b[0mmyframe2\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNameError\u001b[0m: name 'myframe2' is not defined"
     ]
    }
   ],
   "source": [
    "luckynumber = ['3','2','1','7','8']\n",
    "luckynumber = pd.Series(luckynumber)\n",
    "myframe2['luckynumber'] = luckynumber\n",
    "myframe2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "96f8c520",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tf.Tensor(\n",
      "[[ 6.  8.]\n",
      " [10. 12.]], shape=(2, 2), dtype=float32)\n"
     ]
    }
   ],
   "source": [
    "import tensorflow as tf\n",
    "\n",
    "# 啟用即時執行模式\n",
    "tf.compat.v1.enable_eager_execution()\n",
    "\n",
    "# 創建一個Tensor\n",
    "a = tf.constant([[1.0, 2.0], [3.0, 4.0]])\n",
    "\n",
    "# 創建另一個Tensor\n",
    "b = tf.constant([[5.0, 6.0], [7.0, 8.0]])\n",
    "\n",
    "# 相加\n",
    "c = tf.add(a, b)\n",
    "\n",
    "print(c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "23caac60",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-44-a35612fe959a>\", line 4, in <module>\n",
      "    init_op=tf.global_variables_initializer()\n",
      "AttributeError: module 'tensorflow' has no attribute 'global_variables_initializer'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-44-a35612fe959a>\", line 4, in <module>\n",
      "    init_op=tf.global_variables_initializer()\n",
      "AttributeError: module 'tensorflow' has no attribute 'global_variables_initializer'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3263, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3360, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1193, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-44-a35612fe959a>\", line 4, in <module>\n",
      "    init_op=tf.global_variables_initializer()\n",
      "AttributeError: module 'tensorflow' has no attribute 'global_variables_initializer'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3263, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3360, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1193, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2895, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3072, in run_cell_async\n",
      "    interactivity=interactivity, compiler=compiler, result=result)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3282, in run_ast_nodes\n",
      "    self.showtraceback()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1211, in structured_traceback\n",
      "    chained_exceptions_tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n"
     ]
    }
   ],
   "source": [
    "import tensorflow as tf\n",
    "tf.compat.v1.enable_eager_execution()\n",
    "b=tf.Variable(tf.zeros([2]))\n",
    "init_op=tf.global_variables_initializer()\n",
    "print(init_op)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d64abcad",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR! Session/line number was not unique in database. History logging moved to new session 64\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-44-48e3cf1d02a2>\", line 17, in <module>\n",
      "    result = sess.run(c)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow_core\\python\\client\\session.py\", line 960, in run\n",
      "    run_metadata_ptr)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow_core\\python\\client\\session.py\", line 1108, in _run\n",
      "    raise RuntimeError('The Session graph is empty.  Add operations to the '\n",
      "RuntimeError: The Session graph is empty.  Add operations to the graph before calling run().\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'RuntimeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-44-48e3cf1d02a2>\", line 17, in <module>\n",
      "    result = sess.run(c)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow_core\\python\\client\\session.py\", line 960, in run\n",
      "    run_metadata_ptr)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow_core\\python\\client\\session.py\", line 1108, in _run\n",
      "    raise RuntimeError('The Session graph is empty.  Add operations to the '\n",
      "RuntimeError: The Session graph is empty.  Add operations to the graph before calling run().\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'RuntimeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3263, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3360, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1193, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-44-48e3cf1d02a2>\", line 17, in <module>\n",
      "    result = sess.run(c)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow_core\\python\\client\\session.py\", line 960, in run\n",
      "    run_metadata_ptr)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow_core\\python\\client\\session.py\", line 1108, in _run\n",
      "    raise RuntimeError('The Session graph is empty.  Add operations to the '\n",
      "RuntimeError: The Session graph is empty.  Add operations to the graph before calling run().\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'RuntimeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3263, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3360, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1193, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2895, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3072, in run_cell_async\n",
      "    interactivity=interactivity, compiler=compiler, result=result)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3282, in run_ast_nodes\n",
      "    self.showtraceback()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1211, in structured_traceback\n",
      "    chained_exceptions_tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n"
     ]
    }
   ],
   "source": [
    "#deprecate\n",
    "import tensorflow as tf\n",
    "\n",
    "# 創建一個Session對象\n",
    "sess = tf.compat.v1.Session()\n",
    "\n",
    "# 創建一個Tensor\n",
    "a = tf.constant([[1.0, 2.0], [3.0, 4.0]])\n",
    "\n",
    "# 創建另一個Tensor\n",
    "b = tf.constant([[5.0, 6.0], [7.0, 8.0]])\n",
    "\n",
    "# 相加\n",
    "c = tf.add(a, b)\n",
    "\n",
    "# 使用Session計算結果\n",
    "result = sess.run(c)\n",
    "\n",
    "# 關閉Session\n",
    "sess.close()\n",
    "\n",
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca6609ea",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-44-ff1381a3afee>\", line 4, in <module>\n",
      "    with tf.Session() as sess:\n",
      "AttributeError: module 'tensorflow' has no attribute 'Session'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "ERROR! Session/line number was not unique in database. History logging moved to new session 65\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-44-ff1381a3afee>\", line 4, in <module>\n",
      "    with tf.Session() as sess:\n",
      "AttributeError: module 'tensorflow' has no attribute 'Session'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3263, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3360, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1193, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-44-ff1381a3afee>\", line 4, in <module>\n",
      "    with tf.Session() as sess:\n",
      "AttributeError: module 'tensorflow' has no attribute 'Session'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3263, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3360, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1193, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2895, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3072, in run_cell_async\n",
      "    interactivity=interactivity, compiler=compiler, result=result)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3282, in run_ast_nodes\n",
      "    self.showtraceback()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1211, in structured_traceback\n",
      "    chained_exceptions_tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 953, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n"
     ]
    }
   ],
   "source": [
    "# deprecate\n",
    "import tensorflow as tf\n",
    "result=tf.constant(5)\n",
    "with tf.Session() as sess:\n",
    "    result=sess.run(result)\n",
    "    print (result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55374540",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.datasets import load_diabetes # 匯入Sklearn內建資料集\n",
    "data = load_diabetes() # 將數據匯入\n",
    "import pandas as pd\n",
    "df = pd.DataFrame(data.data,columns = data.feature_names) \n",
    "# 透過data.data來呼叫數據　回傳numpy.ndarray型態\n",
    "# 透過data.feature_names來呼叫特徵名稱\n",
    "# 將原先的data由numpy.ndarray變更為pandas.DataFrame型態\n",
    "# 將特徵名稱與資料放入對應的行(columns)\n",
    "\n",
    "df.head() # 顯示頭五筆資料\n",
    "# 可使用data.head(10)顯示前10筆或是使用data.tail()顯示最後五筆資料\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ca616834",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0a3feced",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn import datasets\n",
    "data = datasets() # 將數據匯入\n",
    "import pandas as pd\n",
    "df = pd.DataFrame(data.data,columns = data.feature_names)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "aa89ba8d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "data (150, 4)\n",
      "target (150,)\n",
      "frame\n",
      "target_names (3,)\n",
      "DESCR\n",
      "feature_names\n",
      "filename\n",
      "/////\n",
      "['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']\n",
      "/////\n",
      ".. _iris_dataset:\n",
      "\n",
      "Iris plants dataset\n",
      "--------------------\n",
      "\n",
      "**Data Set Characteristics:**\n",
      "\n",
      "    :Number of Instances: 150 (50 in each of three classes)\n",
      "    :Number of Attributes: 4 numeric, predictive attributes and the class\n",
      "    :Attribute Information:\n",
      "        - sepal length in cm\n",
      "        - sepal width in cm\n",
      "        - petal length in cm\n",
      "        - petal width in cm\n",
      "        - class:\n",
      "                - Iris-Setosa\n",
      "                - Iris-Versicolour\n",
      "                - Iris-Virginica\n",
      "                \n",
      "    :Summary Statistics:\n",
      "\n",
      "    ============== ==== ==== ======= ===== ====================\n",
      "                    Min  Max   Mean    SD   Class Correlation\n",
      "    ============== ==== ==== ======= ===== ====================\n",
      "    sepal length:   4.3  7.9   5.84   0.83    0.7826\n",
      "    sepal width:    2.0  4.4   3.05   0.43   -0.4194\n",
      "    petal length:   1.0  6.9   3.76   1.76    0.9490  (high!)\n",
      "    petal width:    0.1  2.5   1.20   0.76    0.9565  (high!)\n",
      "    ============== ==== ==== ======= ===== ====================\n",
      "\n",
      "    :Missing Attribute Values: None\n",
      "    :Class Distribution: 33.3% for each of 3 classes.\n",
      "    :Creator: R.A. Fisher\n",
      "    :Donor: Michael Marshall (MARSHALL%PLU@io.arc.nasa.gov)\n",
      "    :Date: July, 1988\n",
      "\n",
      "The famous Iris database, first used by Sir R.A. Fisher. The dataset is taken\n",
      "from Fisher's paper. Note that it's the same as in R, but not as in the UCI\n",
      "Machine Learning Repository, which has two wrong data points.\n",
      "\n",
      "This is perhaps the best known database to be found in the\n",
      "pattern recognition literature.  Fisher's paper is a classic in the field and\n",
      "is referenced frequently to this day.  (See Duda & Hart, for example.)  The\n",
      "data set contains 3 classes of 50 instances each, where each class refers to a\n",
      "type of iris plant.  One class is linearly separable from the other 2; the\n",
      "latter are NOT linearly separable from each other.\n",
      "\n",
      ".. topic:: References\n",
      "\n",
      "   - Fisher, R.A. \"The use of multiple measurements in taxonomic problems\"\n",
      "     Annual Eugenics, 7, Part II, 179-188 (1936); also in \"Contributions to\n",
      "     Mathematical Statistics\" (John Wiley, NY, 1950).\n",
      "   - Duda, R.O., & Hart, P.E. (1973) Pattern Classification and Scene Analysis.\n",
      "     (Q327.D83) John Wiley & Sons.  ISBN 0-471-22361-1.  See page 218.\n",
      "   - Dasarathy, B.V. (1980) \"Nosing Around the Neighborhood: A New System\n",
      "     Structure and Classification Rule for Recognition in Partially Exposed\n",
      "     Environments\".  IEEE Transactions on Pattern Analysis and Machine\n",
      "     Intelligence, Vol. PAMI-2, No. 1, 67-71.\n",
      "   - Gates, G.W. (1972) \"The Reduced Nearest Neighbor Rule\".  IEEE Transactions\n",
      "     on Information Theory, May 1972, 431-433.\n",
      "   - See also: 1988 MLC Proceedings, 54-64.  Cheeseman et al\"s AUTOCLASS II\n",
      "     conceptual clustering system finds 3 classes in the data.\n",
      "   - Many, many more ...\n"
     ]
    }
   ],
   "source": [
    "from sklearn import datasets\n",
    "iris = datasets.load_iris()\n",
    "for key,value in iris.items() :\n",
    "    try:\n",
    "        print (key,value.shape)\n",
    "    except:\n",
    "        print (key)\n",
    "print (\"/////\")\n",
    "print (iris['feature_names']) #鳶尾花的特徵\n",
    "print (\"/////\")\n",
    "print (iris['DESCR'])  #英翻中當然就是檔案描述囉"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "dcc5505c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.050680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>-0.044642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.050680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>-0.044642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>-0.044642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>437</th>\n",
       "      <td>0.050680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>438</th>\n",
       "      <td>0.050680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>0.050680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>-0.044642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>-0.044642</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>442 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          sex\n",
       "0    0.050680\n",
       "1   -0.044642\n",
       "2    0.050680\n",
       "3   -0.044642\n",
       "4   -0.044642\n",
       "..        ...\n",
       "437  0.050680\n",
       "438  0.050680\n",
       "439  0.050680\n",
       "440 -0.044642\n",
       "441 -0.044642\n",
       "\n",
       "[442 rows x 1 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[0,\"sex\"] # ->回傳0.050680\n",
    "df.loc[0,\"age\"] # ->回傳-0.001882\n",
    "df.loc[:,[\"sex\"]] #將回傳DataFrame型態"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "04bd8f89",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      0.050680\n",
       "1     -0.044642\n",
       "2      0.050680\n",
       "3     -0.044642\n",
       "4     -0.044642\n",
       "         ...   \n",
       "437    0.050680\n",
       "438    0.050680\n",
       "439    0.050680\n",
       "440   -0.044642\n",
       "441   -0.044642\n",
       "Name: sex, Length: 442, dtype: float64"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[:,\"sex\"] #將回傳Series型態"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "9f26571e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      1\n",
       "1      0\n",
       "2      1\n",
       "3      0\n",
       "4      0\n",
       "      ..\n",
       "437    1\n",
       "438    1\n",
       "439    1\n",
       "440    0\n",
       "441    0\n",
       "Name: sex, Length: 442, dtype: int64"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "condition = lambda x: 1 if x > 0 else 0\n",
    "df.loc[:,\"sex\"].apply(condition)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "3954a50d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 442 entries, 0 to 441\n",
      "Data columns (total 10 columns):\n",
      " #   Column  Non-Null Count  Dtype  \n",
      "---  ------  --------------  -----  \n",
      " 0   age     442 non-null    float64\n",
      " 1   sex     442 non-null    float64\n",
      " 2   bmi     442 non-null    float64\n",
      " 3   bp      442 non-null    float64\n",
      " 4   s1      442 non-null    float64\n",
      " 5   s2      442 non-null    float64\n",
      " 6   s3      442 non-null    float64\n",
      " 7   s4      442 non-null    float64\n",
      " 8   s5      442 non-null    float64\n",
      " 9   s6      442 non-null    float64\n",
      "dtypes: float64(10)\n",
      "memory usage: 34.7 KB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "855896b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[2, 3, 4],\n",
       "       [5, 6, 7]])"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = np.array([1, 2, 3]) # 建立一維陣列\n",
    "b = np.array([(2, 3, 4), (5, 6, 7)]) # 建立二維陣列\n",
    "a\n",
    "b"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "b6575a8a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([1, 3, 5, 7])"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c = np.arange(10) # 0到9，不包含10\n",
    "d = np.arange(1, 8, 2) # 1到7，間隔為2\n",
    "d"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "47a607e8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([6.36598737e-314, 1.48539705e-313])"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "e = np.zeros(3) # 全是0,dtype是float\n",
    "f = np.ones(3) # 全是1,dtype是float\n",
    "g = np.empty(2) # 空的，但非零，有殘留值\n",
    "g"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "95a44c86",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'='"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dt = np.dtype('i2')\n",
    "\n",
    "dt.byteorder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "471243d0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'|'"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.dtype('i1').byteorder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c4aa66ea",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'|'"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.dtype('S2').byteorder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "81092d27",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'='"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import sys\n",
    "\n",
    "sys_is_le = sys.byteorder == 'little'\n",
    "\n",
    "native_code = sys_is_le and '<' or '>'\n",
    "\n",
    "swapped_code = sys_is_le and '>' or '<'\n",
    "\n",
    "dt = np.dtype(native_code + 'i2')\n",
    "\n",
    "dt.byteorder\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "9b1de9fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dt = np.dtype(swapped_code + 'i2')\n",
    "\n",
    "dt.byteorder == swapped_code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "a2917a21",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'>'"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "swapped_code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "499495b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a=1\n",
    "b = a and 3 or 2\n",
    "b"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "de67833e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[10. 12. 14. 16. 18.]\n",
      "The step size is 2.0\n",
      "[1.00000000e+010 3.16227766e+257             inf             inf\n",
      "             inf]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\johnn\\anaconda3\\envs\\tf2\\lib\\site-packages\\numpy\\core\\function_base.py:265: RuntimeWarning: overflow encountered in power\n",
      "  return _nx.power(base, y)\n"
     ]
    }
   ],
   "source": [
    "arr, step_size = np.linspace(10, 20,  5, endpoint =  False, retstep=True) # 10到20(不包含)等差的5個數\n",
    "i = np.logspace(10, 1000,  5, endpoint =  True) # 10到1000(包含)等比(以10為底)的五個數\n",
    "print(arr)\n",
    "print('The step size is ' + str(step_size))\n",
    "print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "c37e14ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Team</th>\n",
       "      <th>Number</th>\n",
       "      <th>Position</th>\n",
       "      <th>Age</th>\n",
       "      <th>Height</th>\n",
       "      <th>Weight</th>\n",
       "      <th>College</th>\n",
       "      <th>Salary</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Avery Bradley</td>\n",
       "      <td>Boston Celtics</td>\n",
       "      <td>0.0</td>\n",
       "      <td>PG</td>\n",
       "      <td>25.0</td>\n",
       "      <td>6-2</td>\n",
       "      <td>180.0</td>\n",
       "      <td>Texas</td>\n",
       "      <td>7730337.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Jae Crowder</td>\n",
       "      <td>Boston Celtics</td>\n",
       "      <td>99.0</td>\n",
       "      <td>SF</td>\n",
       "      <td>25.0</td>\n",
       "      <td>6-6</td>\n",
       "      <td>235.0</td>\n",
       "      <td>Marquette</td>\n",
       "      <td>6796117.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>John Holland</td>\n",
       "      <td>Boston Celtics</td>\n",
       "      <td>30.0</td>\n",
       "      <td>SG</td>\n",
       "      <td>27.0</td>\n",
       "      <td>6-5</td>\n",
       "      <td>205.0</td>\n",
       "      <td>Boston University</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Name            Team  Number Position   Age Height  Weight  \\\n",
       "0  Avery Bradley  Boston Celtics     0.0       PG  25.0    6-2   180.0   \n",
       "1    Jae Crowder  Boston Celtics    99.0       SF  25.0    6-6   235.0   \n",
       "2   John Holland  Boston Celtics    30.0       SG  27.0    6-5   205.0   \n",
       "\n",
       "             College     Salary  \n",
       "0              Texas  7730337.0  \n",
       "1          Marquette  6796117.0  \n",
       "2  Boston University        NaN  "
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_csv('./nba.csv')\n",
    "df.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d9ecb428",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TensorFlow version: 2.9.1\n"
     ]
    }
   ],
   "source": [
    "import tensorflow as tf\n",
    "print(\"TensorFlow version:\", tf.__version__)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "432564eb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloading data from https://storage.googleapis.com/tensorflow/tf-keras-datasets/mnist.npz\n",
      "11490434/11490434 [==============================] - 4s 0us/step\n"
     ]
    }
   ],
   "source": [
    "mnist = tf.keras.datasets.mnist\n",
    "\n",
    "(x_train, y_train), (x_test, y_test) = mnist.load_data()\n",
    "x_train, x_test = x_train / 255.0, x_test / 255.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "3378b8b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "model = tf.keras.models.Sequential([\n",
    "  tf.keras.layers.Flatten(input_shape=(28, 28)),\n",
    "  tf.keras.layers.Dense(128, activation='relu'),\n",
    "  tf.keras.layers.Dropout(0.2),\n",
    "  tf.keras.layers.Dense(10)\n",
    "])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "475752d5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Iteration 1: x = 0.4, f(x) = -0.4399999999999995\n",
      "Iteration 2: x = 0.7200000000000001, f(x) = -1.3616000000000006\n",
      "Iteration 3: x = 0.976, f(x) = -1.951424\n",
      "Iteration 4: x = 1.1808, f(x) = -2.32891136\n",
      "Iteration 5: x = 1.34464, f(x) = -2.5705032704\n",
      "Iteration 6: x = 1.4757120000000001, f(x) = -2.725122093056\n",
      "Iteration 7: x = 1.5805696, f(x) = -2.82407813955584\n",
      "Iteration 8: x = 1.66445568, f(x) = -2.8874100093157375\n",
      "Iteration 9: x = 1.731564544, f(x) = -2.9279424059620722\n",
      "Iteration 10: x = 1.7852516352, f(x) = -2.9538831398157264\n",
      "Iteration 11: x = 1.8282013081600001, f(x) = -2.9704852094820646\n",
      "Iteration 12: x = 1.862561046528, f(x) = -2.9811105340685216\n",
      "Iteration 13: x = 1.8900488372224, f(x) = -2.9879107418038537\n",
      "Iteration 14: x = 1.91203906977792, f(x) = -2.9922628747544664\n",
      "Iteration 15: x = 1.9296312558223359, f(x) = -2.9950482398428586\n",
      "Iteration 16: x = 1.9437050046578688, f(x) = -2.9968308734994293\n",
      "Iteration 17: x = 1.954964003726295, f(x) = -2.997971759039635\n",
      "Iteration 18: x = 1.963971202981036, f(x) = -2.998701925785366\n",
      "Iteration 19: x = 1.9711769623848288, f(x) = -2.9991692325026342\n",
      "Iteration 20: x = 1.976941569907863, f(x) = -2.9994683088016862\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "1.976941569907863"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "\n",
    "def function(x):\n",
    "    # Define the function we are trying to minimize\n",
    "    return (x-2)**2 - 3\n",
    "\n",
    "def derivative(x):\n",
    "    # The derivative (gradient) of the function\n",
    "    return 2*(x-2)\n",
    "\n",
    "def gradient_descent(x_start, learning_rate, n_iters):\n",
    "    x = x_start\n",
    "    for i in range(n_iters):\n",
    "        grad = derivative(x) # Compute the gradient at the current location\n",
    "        x = x - learning_rate * grad # Update the location in the direction of the gradient\n",
    "        print(f\"Iteration {i+1}: x = {x}, f(x) = {function(x)}\")\n",
    "    return x\n",
    "\n",
    "# Run gradient descent starting from x=0, with learning rate 0.1, for 20 iterations\n",
    "gradient_descent(x_start=0, learning_rate=0.1, n_iters=20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "ff063b39",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "size 44\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAk0AAAGwCAYAAAC0HlECAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8qNh9FAAAACXBIWXMAAA9hAAAPYQGoP6dpAABDXElEQVR4nO3de3RU5b3/8c8k5EYIuwQaJkHUiGhNQ1ViuWkLiEBUQm+rtnBI6dHiT1CQClWpXQVsC1it9kKLl56jtVjTVtRTKo1gBSwlEAzGElOtYpCLiVESJgHJhcnz+4Nml8mF7Jlc5vZ+rZW1mr2/mTzDrsyHZz/7+7iMMUYAAAA4q5hgDwAAACAcEJoAAAAcIDQBAAA4QGgCAABwgNAEAADgAKEJAADAAUITAACAA/2CPYBI0tLSovfff18pKSlyuVzBHg4AAHDAGKP6+nplZGQoJqbz+SRCUw96//33NXz48GAPAwAABODQoUM655xzOj1PaOpBKSkpkk7/oQ8cODDIowEAAE7U1dVp+PDh9ud4ZwhNPaj1ltzAgQMJTQAAhJmultawEBwAAMCBoIamFStWyOVy+Xy53W77vDFGK1asUEZGhpKSkjRp0iS98cYbPq/R2NiohQsXasiQIUpOTtbMmTN1+PBhn5ra2lrl5+fLsixZlqX8/HwdO3bMp+bgwYPKy8tTcnKyhgwZokWLFqmpqanX3jsAAAgvQZ9p+vSnP63Kykr7a9++ffa5H//4x3rwwQe1du1a7dmzR263W1OnTlV9fb1ds3jxYj333HMqKCjQjh07dPz4cc2YMUNer9eumT17tkpLS1VYWKjCwkKVlpYqPz/fPu/1enX99dfrxIkT2rFjhwoKCrRhwwYtWbKkb/4QAABA6DNBtHz5cnPppZd2eK6lpcW43W6zZs0a+1hDQ4OxLMs8/PDDxhhjjh07ZuLi4kxBQYFdc+TIERMTE2MKCwuNMcaUl5cbSWbXrl12TVFRkZFk3nzzTWOMMZs2bTIxMTHmyJEjds3TTz9tEhISjMfj6XT8DQ0NxuPx2F+HDh0yks76MwAAILR4PB5Hn99Bn2l6++23lZGRoczMTH3961/Xu+++K0mqqKhQVVWVpk2bZtcmJCRo4sSJ2rlzpySppKREzc3NPjUZGRnKzs62a4qKimRZlsaOHWvXjBs3TpZl+dRkZ2crIyPDrpk+fboaGxtVUlLS6dhXr15t3/KzLIt2AwAARLCghqaxY8fqySef1IsvvqjHHntMVVVVmjBhgo4ePaqqqipJ0tChQ31+ZujQofa5qqoqxcfHa9CgQWetSUtLa/e709LSfGra/p5BgwYpPj7erunIsmXL5PF47K9Dhw75+ScAAADCRVBbDlx77bX2/x41apTGjx+vESNG6De/+Y3GjRsnqf3jf8aYLh8JbFvTUX0gNW0lJCQoISHhrGMBAACRIei3586UnJysUaNG6e2337afoms701NdXW3PCrndbjU1Nam2tvasNR988EG73/Xhhx/61LT9PbW1tWpubm43AwUAAKJTSIWmxsZG/fOf/1R6eroyMzPldru1ZcsW+3xTU5O2b9+uCRMmSJJycnIUFxfnU1NZWamysjK7Zvz48fJ4PCouLrZrdu/eLY/H41NTVlamyspKu2bz5s1KSEhQTk5Or75nAAAQHoJ6e27p0qXKy8vTueeeq+rqav3whz9UXV2d5s6dK5fLpcWLF2vVqlUaOXKkRo4cqVWrVql///6aPXu2JMmyLN10001asmSJBg8erNTUVC1dulSjRo3SNddcI0m65JJLlJubq3nz5umRRx6RJN18882aMWOGLr74YknStGnTlJWVpfz8fN1///2qqanR0qVLNW/ePDp7AwAQZN4Wo+KKGlXXNygtJVFjMlMVG3P2pTq9Iaih6fDhw5o1a5Y++ugjffKTn9S4ceO0a9cunXfeeZKkO++8UydPntSCBQtUW1ursWPHavPmzT57wzz00EPq16+fbrjhBp08eVJTpkzRE088odjYWLvmqaee0qJFi+yn7GbOnKm1a9fa52NjY/XCCy9owYIFuvLKK5WUlKTZs2frgQce6KM/CQAA0JHCskqt3FiuSk+DfSzdStTyvCzlZqf36VhcxhjTp78xgtXV1cmyLHk8HmaoAADoQlczSIVllZq/fq/aBpXWinVzRvdIcHL6+c2GvQAAoM91NYPkbTFaubG8XWCSJKPTwWnlxnJNzXL32a26kFoIDgAAIl/rDNKZgUmSqjwNmr9+rwrLKlVcUdPu/JmMpEpPg4oranp5tP9BaAIAAH2mqxkk6fQMUlVd54HpTNX1zup6AqEJAAD0GaczSDXHGx29XlpKYg+NrGuEJgAA0GeczgylJscr3UpUZ6uVXDq9BmpMZmqPja0rhCYAANBnnM4Mua0kLc/LkqR2wan1++V5WX3ar4nQBAAA+syYzFTHM0i52elaN2e03JZv0HJbiT3WbsAftBwAAAB9JjbGpeV5WZq/fq9cks+C8I5mkHKz0zU1yx0SHcFpbtmDaG4JAIh2Trc8CaVO3zS3BAAAfcqfIBRKM0hOMdPUg5hpAgBEq77a8qQ3OP38ZiE4AADoFqcNK70t4T1PQ2gCAADdEopbnvQGQhMAAOgWpw0r+3LLk95AaAIAAN3itGFlX2550hsITQAAoFv8aVgZzghNAACgW1obVkqhs+VJbyA0AQCAbgu1LU96A80tAQBAjwjHhpX+IDQBAIAuOd0eJTbGpfEjBgdhhL2P0AQAQJQKx33igonQBABAhHEShpwGoc62R6nyNGj++r0Rs17JCUITAAARxEkYchqEutoexaXT26NMzXJHzLqls+HpOQAAIkRrGGq7pUlrGCosq/Rrn7ho2R7FKWaaAACIAE5nhVIS4xwHoWjZHsUpZpoAAIgATmeFivYfdfR6reuhnAj37VGcIjQBABABnM/2dDQX1V7rAvJo2B7FKUITAAARwOlsz/gLhjgOQtGyPYpThCYAABzythgV7T+q/ys9oqL9R+Vt6XjWxmldT3I6KzRuxGC/glA0bI/ilMsY0/tXMkrU1dXJsix5PB4NHDgw2MMBAPQgf/oa9UYjSKe9l+av3yvJ9yZca9WZIcffcTpthBmOnH5+E5p6EKEJACJTZ32N2oYRp3WB/H6nAcef2kgOQv4gNAUBoQkAIo+3xeiq+17u9Mk0l07fqtr+ncmaeP/WLut23HW1X8EkkCBGGPKP089v+jQBAHAWTh/l/23RAcf9j5xuaBtoR+5I3jQ3mFgIDgDAWTh9lP+9mo979PUk54EtWjpyBxuhCQCAs3D6KP95qf179PUk5wErWjpyBxuhCQAQ1bpqD+D0Uf788ef3eCNIOnKHFtY0AQCilpMnzVobPM5fv1cudfwo//K8LMX3i3FU58+C7NbAVuVp6HBdU+vi8mjpyB1szDQBACJSVzNIrU+ltV0zVOVp0Pz1e1VYVmkfc9rgsacbQdKRO7TQcqAH0XIAAEJDVzNITtsItG0P4PRR/p5+5L+3GmbiNPo0BQGhCQCCz0lfIyspXrMe29Xlaz09b1zIPLpP76XeQ58mAEDUcdrX6M7cTzl6vVB6Ko3eS8FHaAIAhI2uZluc9jWqOd7o6PfxVBrORGgCAIQFJ+t6nM4MpSbH81Qa/MbTcwCAkOf0STenM0NuK4mn0uA3QhMAIKR1tU5JOr1OydtiHDeiHJOZ2uPtARD5uD0HAAi6s61V8mf/tfEjBvvVYDI3O11Ts9xBfSqNp+LCB6EJAODXB3df9yDyd/+11hmktq/p7qSvUTCfSqP/UnghNAFAlPPng7unP+Q766nUulZp3ZzRAe2/FgozSF1x8t4JTqGFNU0AEMX82UrEn1onnK5VyjlvUEAb4bbOIH3hsmEaP2JwSAUmf9ZpIXQQmgAgSvnzwR3Ih3xXe785XatU8l5txD3p5s86LYQObs8BQJTy94Pbn8XYPdlTqbq+QV+4bJhf65RCnb/rtBAaCE0AEKV644O7ur7B8Vodf9cqhcM6JacCWaeF4CM0AUCU6o0P7iHJCVr6zOtd7v02Nctt91Typyt3pOy/Fsh7R/CxpgkAItjZ1hX50wjSaa1czm/jxca4Im6tklPR/N7DGaEJACJUYVmlrrrvZc16bJduLyjVrMd26ar7XrafcvPng9tp7UcON8Jt21MpGrtyR/N7D1cuYwzPM/aQuro6WZYlj8ejgQMHBns4AKJYZ+uKWgPOmR/KPdmnqWj/Uc16bFeX43t63jif22zR3BU7mt97qHD6+U1o6kGEJgChwNtidNV9L3d6m6x1vcyOu662P5x7qiN46+/uaq3Omb8bCDann98sBAeACOPvXm2Sfwusz1bbehvP6d5vQDhhTRMARJhg9wBirQ4iFTNNABBhQqEHUCT1VAJaEZoAIMKESg+gSOmpBLQKmdtzq1evlsvl0uLFi+1jxhitWLFCGRkZSkpK0qRJk/TGG2/4/FxjY6MWLlyoIUOGKDk5WTNnztThw4d9ampra5Wfny/LsmRZlvLz83Xs2DGfmoMHDyovL0/JyckaMmSIFi1apKampt56uwDQa+gBBPSOkAhNe/bs0aOPPqrPfOYzPsd//OMf68EHH9TatWu1Z88eud1uTZ06VfX19XbN4sWL9dxzz6mgoEA7duzQ8ePHNWPGDHm9Xrtm9uzZKi0tVWFhoQoLC1VaWqr8/Hz7vNfr1fXXX68TJ05ox44dKigo0IYNG7RkyZLef/MA4KeuNsKVWFcE9AoTZPX19WbkyJFmy5YtZuLEieb22283xhjT0tJi3G63WbNmjV3b0NBgLMsyDz/8sDHGmGPHjpm4uDhTUFBg1xw5csTExMSYwsJCY4wx5eXlRpLZtWuXXVNUVGQkmTfffNMYY8ymTZtMTEyMOXLkiF3z9NNPm4SEBOPxeBy/F4/HYyT59TMA4I+/7HvfjFv1kjnvrj/bX+NWvWT+su/9DutPeVvMznc+Ms+/dtjsfOcjc8rb0scjBkKf08/voM803Xrrrbr++ut1zTXX+ByvqKhQVVWVpk2bZh9LSEjQxIkTtXPnTklSSUmJmpubfWoyMjKUnZ1t1xQVFcmyLI0dO9auGTdunCzL8qnJzs5WRkaGXTN9+nQ1NjaqpKSk07E3Njaqrq7O5wsAektrw8q27QRaN8Jt7fR9ptZ1RV+4bJjGjxjMLTmgG4IamgoKClRSUqLVq1e3O1dVVSVJGjp0qM/xoUOH2ueqqqoUHx+vQYMGnbUmLS2t3eunpaX51LT9PYMGDVJ8fLxd05HVq1fb66Qsy9Lw4cO7essAEBBvi9HKjeWdboQrnd4It6NbdQB6RtBC06FDh3T77bfrqaeeUmJi54+9uly+/yoyxrQ71lbbmo7qA6lpa9myZfJ4PPbXoUOHzjouAAiUPw0rAfSOoIWmkpISVVdXKycnR/369VO/fv20fft2/fznP1e/fv3smZ+2Mz3V1dX2ObfbraamJtXW1p615oMPPmj3+z/88EOfmra/p7a2Vs3Nze1moM6UkJCggQMH+nwBQG8IdsNKAEEMTVOmTNG+fftUWlpqf11xxRX6r//6L5WWluqCCy6Q2+3Wli1b7J9pamrS9u3bNWHCBElSTk6O4uLifGoqKytVVlZm14wfP14ej0fFxcV2ze7du+XxeHxqysrKVFn5n/UAmzdvVkJCgnJycnr1zwEAnAiFhpVAtAtac8uUlBRlZ2f7HEtOTtbgwYPt44sXL9aqVas0cuRIjRw5UqtWrVL//v01e/ZsSZJlWbrpppu0ZMkSDR48WKmpqVq6dKlGjRplLyy/5JJLlJubq3nz5umRRx6RJN18882aMWOGLr74YknStGnTlJWVpfz8fN1///2qqanR0qVLNW/ePGaPAISEUGlYCUSzkO4Ifuedd+rkyZNasGCBamtrNXbsWG3evFkpKSl2zUMPPaR+/frphhtu0MmTJzVlyhQ98cQTio2NtWueeuopLVq0yH7KbubMmVq7dq19PjY2Vi+88IIWLFigK6+8UklJSZo9e7YeeOCBvnuzAKKet8V0uu0IG+ECwecyxvCoRQ+pq6uTZVnyeDzMUAHwS2FZpVZuLPdZ7J1uJWp5XpZPI0qndQCcc/r5TWjqQYQmAIFo7b/U9i/j1jmjth28zzYjBcB/Tj+/Q/r2HABEuq76L7l0uv/S1Cy3z606NsIF+l7QO4IDQDSj/xIQPghNABBE9F8CwgehCQCCiP5LQPggNAFAELX2X+psGbdLp5+Oo/8SEHyEJgAIotb+S5LaBSf6LwGhhdAEAEGWm52udXNGy2353oJzW4nt2g0ACB5aDgBACMjNTtfULDf9l4AQRmgCgBBB/yUgtHF7DgAAwAFmmgCgF7HlCRA5CE0A0EvYXBeILNyeA4Be0LoJb9stUqo8DZq/fq8KyyqDNDIAgSI0AUAP62oTXun0Jrzelo4qAIQqQhMA9DA24QUiE6EJAHoYm/ACkYnQBAA9jE14gchEaAKAHsYmvEBkIjQBQA9jE14gMhGaAKAXsAkvEHlobgkAAXDS6ZtNeIHIQmgCAD/50+mbTXiByMHtOQDwA52+gehFaAIAh+j0DUQ3QhMAOESnbyC6EZoAwCE6fQPRjdAEAA7R6RuIboQmAHCITt9AdCM0AYBDdPoGohuhCQD8QKdvIHrR3BIA/ESnbyA6EZoAIAB0+gaiD6EJAP7NyX5yAKIXoQkA5N9+cgCiEwvBAUQ99pMD4AShCUBUYz85AE4RmgBENfaTA+AUoQlAVGM/OQBOEZoARDX2kwPgFKEJQFRjPzkAThGaAEQ19pMD4BShCUDUYz85AE7Q3BIAxH5yALpGaAKAf2M/OQBnw+05AAAABwhNAAAADnB7DkDE87YY1ioB6DZCE4CIVlhWqZUby322Skm3ErU8L4un4gD4hdtzACJWYVml5q/f225vuSpPg+av36vCssogjQxAOCI0AYhI3hajlRvLZTo413ps5cZyeVs6qgCA9ghNACJScUVNuxmmMxlJlZ4GFVfU9N2gAIQ1QhOAiFRd33lgCqQOAAhNACJSWkpi10V+1AEAoQlARBqTmap0K7HdJrytXDr9FN2YzNS+HBaAMEZoAhCRYmNcWp6XJUntglPr98vzsujXBMAxQhOAiJWbna51c0bLbfnegnNbiVo3ZzR9mgD4heaWACJabna6pma56QgOoNv8Dk3GGG3fvl1/+9vfdODAAX388cf65Cc/qcsvv1zXXHONhg8f3hvjBICAxca4NH7E4GAPA0CYc3x77uTJk1q1apWGDx+ua6+9Vi+88IKOHTum2NhYvfPOO1q+fLkyMzN13XXXadeuXb05ZgAAgD7neKbpoosu0tixY/Xwww9r+vTpiouLa1fz3nvv6Xe/+52+9rWv6Xvf+57mzZvXo4MFAAAIFsczTX/5y1/0zDPPaMaMGR0GJkk677zztGzZMr399tuaNGlSl6+5bt06feYzn9HAgQM1cOBAjR8/Xn/5y1/s88YYrVixQhkZGUpKStKkSZP0xhtv+LxGY2OjFi5cqCFDhig5OVkzZ87U4cOHfWpqa2uVn58vy7JkWZby8/N17Ngxn5qDBw8qLy9PycnJGjJkiBYtWqSmpiZnfzgAACDiOQ5N2dnZjl80Pj5eI0eO7LLunHPO0Zo1a/Tqq6/q1Vdf1dVXX60vfOELdjD68Y9/rAcffFBr167Vnj175Ha7NXXqVNXX19uvsXjxYj333HMqKCjQjh07dPz4cc2YMUNer9eumT17tkpLS1VYWKjCwkKVlpYqPz/fPu/1enX99dfrxIkT2rFjhwoKCrRhwwYtWbLE8XsGAAARzgTgX//6l7n//vvNrbfeam677Tbzk5/8xOzfvz+Ql2pn0KBB5te//rVpaWkxbrfbrFmzxj7X0NBgLMsyDz/8sDHGmGPHjpm4uDhTUFBg1xw5csTExMSYwsJCY4wx5eXlRpLZtWuXXVNUVGQkmTfffNMYY8ymTZtMTEyMOXLkiF3z9NNPm4SEBOPxeByP3ePxGEl+/QyAwJzytpid73xknn/tsNn5zkfmlLcl2EMCEKacfn77/fTc6tWr9f3vf18tLS1KS0uTMUYffvih7r77bq1atUpLly4NKLx5vV798Y9/1IkTJzR+/HhVVFSoqqpK06ZNs2sSEhI0ceJE7dy5U//v//0/lZSUqLm52acmIyND2dnZ2rlzp6ZPn66ioiJZlqWxY8faNePGjZNlWdq5c6cuvvhiFRUVKTs7WxkZGXbN9OnT1djYqJKSEk2ePLnDMTc2NqqxsdH+vq6uLqD3DsA/hWWVWrmx3GdD3nQrUcvzsui9BKDX+NXccuvWrfre976ne+65Rx999JEqKytVVVVlh6a7775br7zyil8D2LdvnwYMGKCEhATdcssteu6555SVlaWqqipJ0tChQ33qhw4dap+rqqpSfHy8Bg0adNaatLS0dr83LS3Np6bt7xk0aJDi4+Ptmo6sXr3aXidlWRbtFoA+UFhWqfnr9/oEJkmq8jRo/vq9KiyrDNLIAEQ6v0LTww8/rG9961tasWKFT1BJTU3VvffeqxtvvFHr1q3zawAXX3yxSktLtWvXLs2fP19z585VeXm5fd7l8m1AZ4xpd6yttjUd1QdS09ayZcvk8Xjsr0OHDp11XAC6x9titHJjuUwH51qPrdxYLm9LRxUA0D1+habi4mKfBdRt5efn+92jKT4+XhdeeKGuuOIKrV69Wpdeeql+9rOfye12S1K7mZ7q6mp7VsjtdqupqUm1tbVnrfnggw/a/d4PP/zQp6bt76mtrVVzc3O7GagzJSQk2E/+tX4B6D3FFTXtZpjOZCRVehpUXFHTd4MCEDX8Ck0ffPCBzj///E7PZ2ZmnvV2lhPGGDU2NiozM1Nut1tbtmyxzzU1NWn79u2aMGGCJCknJ0dxcXE+NZWVlSorK7Nrxo8fL4/Ho+LiYrtm9+7d8ng8PjVlZWWqrPzPtP7mzZuVkJCgnJycbr0fAD2nur7zwBRIHQD4w6+F4A0NDYqPj+/0fFxcnF+9jb773e/q2muv1fDhw1VfX6+CggJt27ZNhYWFcrlcWrx4sVatWqWRI0dq5MiRWrVqlfr376/Zs2dLkizL0k033aQlS5Zo8ODBSk1N1dKlSzVq1Chdc801kqRLLrlEubm5mjdvnh555BFJ0s0336wZM2bo4osvliRNmzZNWVlZys/P1/3336+amhotXbpU8+bNY/YICCFpKYldF/lRBwD+8PvpuV//+tcaMGBAh+fO7J/kxAcffKD8/HxVVlbKsix95jOfUWFhoaZOnSpJuvPOO3Xy5EktWLBAtbW1Gjt2rDZv3qyUlBT7NR566CH169dPN9xwg06ePKkpU6boiSeeUGxsrF3z1FNPadGiRfZTdjNnztTatWvt87GxsXrhhRe0YMECXXnllUpKStLs2bP1wAMP+PV+APSuMZmpSrcSVeVp6HBdk0uS2zq9IS8A9DSXMcbxisnzzz+/y0XYklRRUdGtQYWruro6WZYlj8fDDBXQS1qfnpPkE5xa/2ZaN2c0bQcA+MXp57dfM00HDhzo7rgAoFtys9O1bs7odn2a3PRpAtDL/ApN77zzji688MLeGgsAOJKbna6pWW4VV9Sour5BaSmnb8nFxnQ9Ew4AgfIrNF100UUaNmyYJk+ebH+d7Wk6AOgtsTEujR8xONjDABBF/ApN27dv1/bt27Vt2zbddtttamho0Lnnnqurr77aDlHDhg3rrbECAAAEjV8Lwc/U3NysoqIibdu2Tdu2bdOuXbvU2NioCy+8UG+99VZPjzMssBAcAIDw4/TzO+DQ1OrkyZPasWOHXnzxRT322GM6fvy4vF5vd14ybBGaAAAIP73y9Jx0usHlzp07tXXrVm3btk179uxRZmamJk6cqHXr1mnixIndGjgAAEAo8is0TZw4UXv27NGIESP0+c9/XgsXLtTEiRPPuj8bAABAJPArNO3cuVPp6emaPHmyJk2apM9//vMaMmRIb40NAAAgZPi1Ye+xY8f06KOPqn///rrvvvs0bNgwjRo1SrfddpueeeYZffjhh701TgBRwttiVLT/qP6v9IiK9h+Vt6Vbyy4BoMd0ayF4fX29duzYYa9vev311zVy5EiVlZX15BjDBgvBge4pLKts1+k7nU7fAHqZ089vv2aa2kpOTlZqaqpSU1M1aNAg9evXT//85z+785IAolTrnnJnBiZJqvI0aP76vSosqwzSyADgNL/WNLW0tOjVV1/Vtm3btHXrVv3973/XiRMn7C7hv/zlLzV58uTeGiuACOVtMVq5sVwdTXsbnd6Md+XGck3NcrNVCoCg8Ss0feITn9CJEyeUnp6uSZMm6cEHH9TkyZM1YsSI3hofgChQXFHTbobpTEZSpadBxRU1bJ0CIGj8Ck3333+/Jk+erIsuuqi3xgMgClXXdx6YAqkDgN7g15qmBQsW6BOf+EQvDQVAtEpLSezROgDoDX6Fpm7uuAIAHRqTmap0K1GdrVZy6fRTdGMyU/tyWADgo1tPzwFAT4iNcWl5XpYktQtOrd8vz8tiETiAoPJ777kXX3xRlmWdtWbmzJkBDwhAdMrNTte6OaPb9Wly06cJQIjwq7llTEzXE1Mul0ter7dbgwpXNLcEus/bYlRcUaPq+galpZy+JccME4De5PTz2++ZpqqqKqWlpXVrcADQmdgYF20FAIQkv9Y0uVz8aw8AAEQnnp4DAABwwK/QNHfuXCUlJfXWWAAAAEKW4zVNJ06c0OOPP+74hU+cOKHk5OSABgUAABBqHM80XXjhhVq1apXef//9TmuMMdqyZYuuvfZa/fznP++RAQIAAIQCxzNN27Zt0/e+9z2tXLlSl112ma644gplZGQoMTFRtbW1Ki8vV1FRkeLi4rRs2TLdfPPNvTluAACAPuVXnyZJOnz4sP74xz/qlVde0YEDB3Ty5EkNGTJEl19+uaZPn67rrrvOUT+nSESfJqBj9F4CEMqcfn77HZrQOUIT0F5hWWW7Lt/pdPkGEEKcfn4HNCW0bdu2QMcFIIoUllVq/vq9PoFJkqo8DZq/fq8KyyqDNDIA8F9AoSk3N1cjRozQD3/4Qx06dKinxwQgAnhbjFZuLFdHU9mtx1ZuLJe3hcluAOEhoND0/vvv6/bbb9ezzz6rzMxMTZ8+XX/4wx/U1NTU0+MDEKaKK2razTCdyUiq9DSouKKm7wYFAN0QUGhKTU3VokWLtHfvXr366qu6+OKLdeuttyo9PV2LFi3S66+/3tPjBBBmqus7D0yB1AFAsHX7MbfLLrtMd999t2699VadOHFC//u//6ucnBx97nOf0xtvvNETYwQQhtJSEnu0DgCCLeDQ1NzcrGeeeUbXXXedzjvvPL344otau3atPvjgA1VUVGj48OH66le/2pNjBRBGxmSmKt1KVGeNBVw6/RTdmMzUvhwWAAQsoNC0cOFCpaen65ZbbtFFF12k1157TUVFRfrWt76l5ORkDR8+XGvWrNGbb77Z0+MFECZiY1xanpclSe2CU+v3y/Oy6NcEIGw47gh+pvLycv3iF7/QV77yFcXHx3dYk5GRoa1bt3ZrcADCW252utbNGd2uT5ObPk0AwpDfoam5uVnnnnuuxo4d22lgkqR+/fpp4sSJ3RocgPCXm52uqVluOoIDCHt+356Li4vTc8891xtjARChYmNcGj9isL5w2TCNHzGYwAQgLAW0pulLX/qSnn/++R4eCgAAQOgKaE3ThRdeqB/84AfauXOncnJylJyc7HN+0aJFPTI4AACAUBHQhr2ZmZmdv6DLpXfffbdbgwpXbNgLAED4cfr5HdBMU0VFRcADAwAACEfd7ggOAAAQDQKaaZKkw4cP609/+pMOHjzYbqPeBx98sNsDAwAACCUBhaa//vWvmjlzpjIzM/XWW28pOztbBw4ckDFGo0eP7ukxAgAABF1At+eWLVumJUuWqKysTImJidqwYYMOHTqkiRMnst8cAACISAGFpn/+85+aO3eupNOdv0+ePKkBAwbo3nvv1X333dejAwQAAAgFAYWm5ORkNTY2Sjq9x9z+/fvtcx999FHPjAwAACCEBLSmady4cfr73/+urKwsXX/99VqyZIn27dunZ599VuPGjevpMQIIUd4Ww55yAKJGQKHpwQcf1PHjxyVJK1as0PHjx/X73/9eF154oR566KEeHSCA0FRYVqmVG8tV6Wmwj6VbiVqel6Xc7PQgjgwAekdAHcHRMTqCI1oUllVq/vq9avuXR+sc07o5owlOAMKG089vmlsC8Iu3xWjlxvJ2gUmSfWzlxnJ5W/j3GIDI4vj23KBBg+RyOVurUFNTE/CAAIS24ooan1tybRlJlZ4GFVfUaPyIwX03MADoZY5D009/+tNeHAaAcFFd33lgCqQOAMKF49DU2pcJQHRLS0ns0ToACBcB7z3X6uTJk2pubvY5xiJoIHKNyUxVupWoKk9Dh+uaXJLc1un2AwAQSQJaCH7ixAnddtttSktL04ABAzRo0CCfLwCRKzbGpeV5WZL+87Rcq9bvl+dl0a8JQMQJKDTdeeedevnll/WrX/1KCQkJ+vWvf62VK1cqIyNDTz75ZE+PEUCIyc1O17o5o+W2fG/Bua1E2g0AiFgB9Wk699xz9eSTT2rSpEkaOHCg9u7dqwsvvFC//e1v9fTTT2vTpk29MdaQR58mRBs6ggOIBE4/vwNa01RTU6PMzExJp9cvtbYYuOqqqzR//vxAXhJAGIqNcdFWAEDUCOj23AUXXKADBw5IkrKysvSHP/xBkrRx40Z94hOf6KmxAQAAhIyAQtN///d/6/XXX5ckLVu2zF7b9O1vf1vf+c53HL/O6tWr9dnPflYpKSlKS0vTF7/4Rb311ls+NcYYrVixQhkZGUpKStKkSZP0xhtv+NQ0NjZq4cKFGjJkiJKTkzVz5kwdPnzYp6a2tlb5+fmyLEuWZSk/P1/Hjh3zqTl48KDy8vKUnJysIUOGaNGiRWpqavLjTwYAAESqHtl77uDBg3r11Vc1YsQIXXrppY5/Ljc3V1//+tf12c9+VqdOndI999yjffv2qby8XMnJyZKk++67Tz/60Y/0xBNP6KKLLtIPf/hDvfLKK3rrrbeUkpIiSZo/f742btyoJ554QoMHD9aSJUtUU1OjkpISxcbGSpKuvfZaHT58WI8++qgk6eabb9b555+vjRs3SpK8Xq8uu+wyffKTn9RPfvITHT16VHPnztWXv/xl/eIXv3D0fljTBABA+HH8+W38sGvXLrNp0yafY7/5zW/M+eefbz75yU+aefPmmYaGBn9e0kd1dbWRZLZv326MMaalpcW43W6zZs0au6ahocFYlmUefvhhY4wxx44dM3FxcaagoMCuOXLkiImJiTGFhYXGGGPKy8uNJLNr1y67pqioyEgyb775pjHGmE2bNpmYmBhz5MgRu+bpp582CQkJxuPxdDjehoYG4/F47K9Dhw4ZSZ3WAwCA0OPxeBx9fvt1e27FihX6xz/+YX+/b98+3XTTTbrmmmu0bNkybdy4UatXrw4k5EmSPB6PJCk19XRTvIqKClVVVWnatGl2TUJCgiZOnKidO3dKkkpKStTc3OxTk5GRoezsbLumqKhIlmVp7Nixds24ceNkWZZPTXZ2tjIyMuya6dOnq7GxUSUlJR2Od/Xq1fbtPsuyNHz48IDfOwAACG1+habS0lJNmTLF/r6goEBjx47VY489pm9/+9v6+c9/bi8K95cxRnfccYeuuuoqZWdnS5KqqqokSUOHDvWpHTp0qH2uqqpK8fHx7Zpqtq1JS0tr9zvT0tJ8atr+nkGDBik+Pt6uaWvZsmXyeDz216FDh/x92wAAIEz41XKgtrbWJ1hs375dubm59vef/exnAw4Ot912m/7xj39ox44d7c65XL59X4wx7Y611bamo/pAas6UkJCghISEs44DAABEBr9mmoYOHaqKigpJUlNTk/bu3avx48fb5+vr6xUXF+f3IBYuXKg//elP2rp1q8455xz7uNvtlqR2Mz3V1dV2eHO73WpqalJtbe1Zaz744IN2v/fDDz/0qWn7e2pra9Xc3NxuBgoAAEQfv0JTbm6u7r77bv3tb3/TsmXL1L9/f33uc5+zz//jH//QiBEjHL+eMUa33Xabnn32Wb388st2w8xWmZmZcrvd2rJli32sqalJ27dv14QJEyRJOTk5iouL86mprKxUWVmZXTN+/Hh5PB4VFxfbNbt375bH4/GpKSsrU2VlpV2zefNmJSQkKCcnx/F7AgAAEcqf1eXV1dXmqquuMi6Xy6SkpJhnn33W5/zVV19tvvvd7zp+vfnz5xvLssy2bdtMZWWl/fXxxx/bNWvWrDGWZZlnn33W7Nu3z8yaNcukp6eburo6u+aWW24x55xzjnnppZfM3r17zdVXX20uvfRSc+rUKbsmNzfXfOYznzFFRUWmqKjIjBo1ysyYMcM+f+rUKZOdnW2mTJli9u7da1566SVzzjnnmNtuu83x+3G6+h4IZae8LWbnOx+Z5187bHa+85E55W0J9pAAoFc5/fwOqE+Tx+PRgAED7B5IrWpqajRgwADFx8c7ep3O1go9/vjj+uY3v9ka6rRy5Uo98sgjqq2t1dixY/XLX/7SXiwuSQ0NDfrOd76j3/3udzp58qSmTJmiX/3qVz5Ps9XU1GjRokX605/+JEmaOXOm1q5d69PB/ODBg1qwYIFefvllJSUlafbs2XrggQccr1uiTxPCXWFZpVZuLFelp8E+lm4lanleFpvwAohYTj+/e6S5JU4jNCGcFZZVav76vWr7F0LrP23WzRlNcAIQkZx+fge0jQqAyOJtMVq5sbxdYJJkH1u5sVzeFv6NBSB6EZoAqLiixueWXFtGUqWnQcUVNX03KAAIMYQmAKqu7zwwBVIHAJGI0ARAaSmJPVoHAJGI0ARAYzJTlW4lqrM++y6dfopuTGZqXw4LAEIKoQmAYmNcWp6XJUntglPr98vzshQbc/btiwAgkhGaAEiScrPTtW7OaLkt31twbiuRdgMAID837AUQ2XKz0zU1y63iihpV1zcoLeX0LTlmmACA0ASgjdgYl8aPGBzsYQBAyOH2HAAAgAOEJgAAAAe4PQdECW+LYa0SAHQDoQmIAoVllVq5sdxnq5R0K1HL87J4Kg4AHOL2HBDhCssqNX/93nZ7y1V5GjR//V4VllUGaWQAEF4ITUAE87YYrdxYLtPBudZjKzeWy9vSUQUA4EyEJiCCFVfUtJthOpORVOlpUHFFTd8NCgDCFKEJiGDV9Z0HpkDqACCaEZqACJaWkth1kR91ABDNCE1ABBuTmap0K7HdJrytXDr9FN2YzNS+HBYAhCVCExDBYmNcWp6XJUntglPr98vzsujXBAAOEJqACJebna51c0bLbfnegnNbiVo3ZzR9mgDAIZpbAmHOSafv3Ox0Tc1y0xEcALqB0ASEMX86fcfGuDR+xOC+HiIARAxuzwFhik7fANC3CE1AGKLTNwD0PUITEIbo9A0AfY/QBIQhOn0DQN8jNAFhiE7fAND3CE1AGKLTNwD0PUITEIK8LUZF+4/q/0qPqGj/0XYLuun0DQB9jz5NQIhx2nuptdN321p3J32aAADd4zLG8ExyD6mrq5NlWfJ4PBo4cGCwh4Mw1Np7qe1/lK3zRR1te+KkIzgAoHNOP7+ZaQJCRFe9l1w63XtpapbbJxTR6RsA+gZrmoA+dLa1SvReAoDQxkwT0Ee6WqtE7yUACG3MNAF9wMk+cfReAoDQRmgCepnTfeJyzhtE7yUACGGEJqCXOV2rVPJeLb2XACCEEZqAXubPWqXW3ktuy/cWnNtK7LDdAACg77AQHOhl/q5Vys1O19QsN72XACDEEJqAXta6T1yVp6HDdU0unZ5JOnOtEr2XACD0cHsO6GXsEwcAkYHQBPQB1ioBQPjj9hzQTU73fmOtEgCEN0IT0A1ddflui7VKABC+uD0HBMhJl28AQOQgNAEBcNrl+8wNeQEA4Y3QBATAaZfv4oqavhsUAKBXsaYJ6MTZFnj70+UbABAZCE1AB7pa4O1vl28AQPjj9hzQhpMF3q1dvjtrFuDS6ZB1ZpdvAEB4IzQBZ3C6wFsSXb4BIMoQmoAz+LPAmy7fABBdWNOEqNJV925/F3jT5RsAogehCVHDSffuQBZ40+UbAKIDt+cQFZx272aBNwCgM4QmRDx/unfHxrhY4A0A6BChCRHP3+7dLPAGAHSENU2IeIF072aBNwCgLUITIl6g3btZ4A0AOBO35xDxWNwNAOgJQQ1Nr7zyivLy8pSRkSGXy6Xnn3/e57wxRitWrFBGRoaSkpI0adIkvfHGGz41jY2NWrhwoYYMGaLk5GTNnDlThw8f9qmpra1Vfn6+LMuSZVnKz8/XsWPHfGoOHjyovLw8JScna8iQIVq0aJGampp6422jj7G4GwDQE4Iamk6cOKFLL71Ua9eu7fD8j3/8Yz344INau3at9uzZI7fbralTp6q+vt6uWbx4sZ577jkVFBRox44dOn78uGbMmCGv12vXzJ49W6WlpSosLFRhYaFKS0uVn59vn/d6vbr++ut14sQJ7dixQwUFBdqwYYOWLFnSe28efYrF3QCAbjMhQpJ57rnn7O9bWlqM2+02a9assY81NDQYy7LMww8/bIwx5tixYyYuLs4UFBTYNUeOHDExMTGmsLDQGGNMeXm5kWR27dpl1xQVFRlJ5s033zTGGLNp0yYTExNjjhw5Ytc8/fTTJiEhwXg8HsfvwePxGEl+/Qz61ilvi9n5zkfm+dcOm53vfGROeVuCPSQAQJA5/fwO2TVNFRUVqqqq0rRp0+xjCQkJmjhxonbu3ClJKikpUXNzs09NRkaGsrOz7ZqioiJZlqWxY8faNePGjZNlWT412dnZysjIsGumT5+uxsZGlZSUdDrGxsZG1dXV+XwhtLUu7v7CZcM0fsRgbskBABwL2dBUVVUlSRo6dKjP8aFDh9rnqqqqFB8fr0GDBp21Ji0trd3rp6Wl+dS0/T2DBg1SfHy8XdOR1atX2+ukLMvS8OHD/XyXAAAgXIRsaGrlcvnOBBhj2h1rq21NR/WB1LS1bNkyeTwe++vQoUNnHRcAAAhfIRua3G63JLWb6amurrZnhdxut5qamlRbW3vWmg8++KDd63/44Yc+NW1/T21trZqbm9vNQJ0pISFBAwcO9PlCcHhbjIr2H9X/lR5R0f6j8rZ0tGkKAACBC9nQlJmZKbfbrS1bttjHmpqatH37dk2YMEGSlJOTo7i4OJ+ayspKlZWV2TXjx4+Xx+NRcXGxXbN79255PB6fmrKyMlVWVto1mzdvVkJCgnJycnr1faL7CssqddV9L2vWY7t0e0GpZj22S1fd97K9CS8AAD0hqB3Bjx8/rnfeecf+vqKiQqWlpUpNTdW5556rxYsXa9WqVRo5cqRGjhypVatWqX///po9e7YkybIs3XTTTVqyZIkGDx6s1NRULV26VKNGjdI111wjSbrkkkuUm5urefPm6ZFHHpEk3XzzzZoxY4YuvvhiSdK0adOUlZWl/Px83X///aqpqdHSpUs1b948Zo9CXGFZpeav39tuM94qT4Pmr99LOwEAQI8Jamh69dVXNXnyZPv7O+64Q5I0d+5cPfHEE7rzzjt18uRJLViwQLW1tRo7dqw2b96slJQU+2ceeugh9evXTzfccINOnjypKVOm6IknnlBsbKxd89RTT2nRokX2U3YzZ8706Q0VGxurF154QQsWLNCVV16ppKQkzZ49Ww888EBv/xGgG7wtRis3lrcLTNLpTXhdklZuLNfULDdPyQEAus1ljGHxRw+pq6uTZVnyeDzMUPWBov1HNeuxXV3WPT1vHHvIAQA65fTzO2TXNAFdqa5v6NE6AADOhtCEsJWWkth1kR91AACcDaEJYWtMZqrSrcR2m/C2cklKtxI1JjO1L4cFAIhQhCaErdgYl5bnZUlSu+DU+v3yvCwWgQMAegShCWEtNztd6+aMltvyvQXnthJpNwAA6FFBbTkA9ITc7HRNzXKruKJG1fUNSks5fUuOGSYAQE8iNCEixMa4aCsAAOhV3J4DAABwgNAEAADgAKEJAADAAUITAACAA4QmAAAAB3h6DiHL22JoIwAACBmEJoSkwrJKrdxYrkrPfzbbTbcStTwvi4aVAICg4PYcQk5hWaXmr9/rE5gkqcrToPnr96qwrDJIIwMARDNCE0KKt8Vo5cZymQ7OtR5bubFc3paOKgAA6D2EJoSU4oqadjNMZzKSKj0NKq6o6btBAQAgQhNCTHV954EpkDoAAHoKoQkhJS0lsUfrAADoKTw9hz53tlYCYzJTlW4lqsrT0OG6Jpckt3X6ZwAA6EuEJvSprloJxMa4tDwvS/PX75VL8glOrR2aludl0a8JANDnuD2HPuO0lUBudrrWzRktt+V7C85tJWrdnNH0aQIABAUzTegTXbUScOl0K4GpWW7FxriUm52uqVluOoIDAEIGoQl9wp9WAuNHDJYkxca47P8NAECwEZrQI7raJ45WAgCAcEdoQrc52SeOVgIAgHDHQnB0i9PF3a2tBDpbkeTS6aBFKwEAQKgiNCFg/uwT19pKQFK74EQrAQBAOCA0IWD+7hNHKwEAQDhjTRPO6mwLvANZ3E0rAQBAuCI0oVNdLfAOdHE3rQQAAOGI23PokJMF3izuBgBEE0IT2nG6wFsSi7sBAFGD0IR2/FngzeJuAEC0YE1TFOrp7t0s7gYARANCU5Tpre7dLO4GAEQ6bs9FEbp3AwAQOEJTlKB7NwAA3UNoiiDeFqOi/Uf1f6VHVLT/qLwt/4lIdO8GAKB7WNMUIbpaq0T3bgAAuofQFAFa1yq1vfXWulZp3ZzRdO8GAKCbuD0X5pyuVco5bxCLuwEA6AZCU5hzulap5L1aFncDANANhKYw589aJRZ3AwAQONY0hbiuunf7u1aJxd0AAASG0BTCnHTvbm1EWeVp6HBdk0unZ5LOXKvE4m4AAPzH7bkQ5bR7N40oAQDoG4SmEORP926JRpQAAPQFbs+FIH+6d7feZmOtEgAAvYvQFIIC6d4tsVYJAIDexO25EBRo924AANB7CE0hqPWJOLp3AwAQOghNIYgn4gAACD2EphDFE3EAAIQWFoKHMJ6IAwAgdBCaQhxPxAEAEBq4PQcAAOAAoQkAAMABQhMAAIADhCYAAAAHCE0AAAAOEJoAAAAcIDS18atf/UqZmZlKTExUTk6O/va3vwV7SAAAIAQQms7w+9//XosXL9Y999yj1157TZ/73Od07bXX6uDBg8EeGgAACDKXMcYEexChYuzYsRo9erTWrVtnH7vkkkv0xS9+UatXr25X39jYqMbGRvv7uro6DR8+XB6PRwMHDuyTMQMAgO6pq6uTZVldfn7TEfzfmpqaVFJSorvvvtvn+LRp07Rz584Of2b16tVauXJlu+N1dXW9MkYAANDzWj+3u5pHIjT920cffSSv16uhQ4f6HB86dKiqqqo6/Jlly5bpjjvusL8/cuSIsrKyNHz48F4dKwAA6Hn19fWyLKvT84SmNlwu381wjTHtjrVKSEhQQkKC/f2AAQNUXl6urKwsHTp0iFt0Iar1NirXKHRxjUIf1yj0cY2cM8aovr5eGRkZZ60jNP3bkCFDFBsb225Wqbq6ut3sU2diYmI0bNgwSdLAgQP5P2mI4xqFPq5R6OMahT6ukTNnm2FqxdNz/xYfH6+cnBxt2bLF5/iWLVs0YcKEII0KAACECmaaznDHHXcoPz9fV1xxhcaPH69HH31UBw8e1C233BLsoQEAgCAjNJ3ha1/7mo4ePap7771XlZWVys7O1qZNm3Teeec5fo2EhAQtX77cZ60TQgvXKPRxjUIf1yj0cY16Hn2aAAAAHGBNEwAAgAOEJgAAAAcITQAAAA4QmgAAABwgNAXoyJEjmjNnjgYPHqz+/fvrsssuU0lJiX3eGKMVK1YoIyNDSUlJmjRpkt54440gjji6nH/++XK5XO2+br31Vklcn1Bw6tQpfe9731NmZqaSkpJ0wQUX6N5771VLS4tdw3UKvvr6ei1evFjnnXeekpKSNGHCBO3Zs8c+zzXqW6+88ory8vKUkZEhl8ul559/3ue8k+vR2NiohQsXasiQIUpOTtbMmTN1+PDhPnwXYczAbzU1Nea8884z3/zmN83u3btNRUWFeemll8w777xj16xZs8akpKSYDRs2mH379pmvfe1rJj093dTV1QVx5NGjurraVFZW2l9btmwxkszWrVuNMVyfUPDDH/7QDB482Pz5z382FRUV5o9//KMZMGCA+elPf2rXcJ2C74YbbjBZWVlm+/bt5u233zbLly83AwcONIcPHzbGcI362qZNm8w999xjNmzYYCSZ5557zue8k+txyy23mGHDhpktW7aYvXv3msmTJ5tLL73UnDp1qo/fTfghNAXgrrvuMldddVWn51taWozb7TZr1qyxjzU0NBjLsszDDz/cF0NEG7fffrsZMWKEaWlp4fqEiOuvv97ceOONPse+/OUvmzlz5hhj+O8oFHz88ccmNjbW/PnPf/Y5fumll5p77rmHaxRkbUOTk+tx7NgxExcXZwoKCuyaI0eOmJiYGFNYWNhnYw9X3J4LwJ/+9CddccUV+upXv6q0tDRdfvnleuyxx+zzFRUVqqqq0rRp0+xjCQkJmjhxonbu3BmMIUe1pqYmrV+/XjfeeKNcLhfXJ0RcddVV+utf/6p//etfkqTXX39dO3bs0HXXXSeJ/45CwalTp+T1epWYmOhzPCkpSTt27OAahRgn16OkpETNzc0+NRkZGcrOzuaaOUBoCsC7776rdevWaeTIkXrxxRd1yy23aNGiRXryySclyd70t+1Gv0OHDm23ITB63/PPP69jx47pm9/8piSuT6i46667NGvWLH3qU59SXFycLr/8ci1evFizZs2SxHUKBSkpKRo/frx+8IMf6P3335fX69X69eu1e/duVVZWco1CjJPrUVVVpfj4eA0aNKjTGnSObVQC0NLSoiuuuEKrVq2SJF1++eV64403tG7dOn3jG9+w61wul8/PGWPaHUPv+5//+R9de+21ysjI8DnO9Qmu3//+91q/fr1+97vf6dOf/rRKS0u1ePFiZWRkaO7cuXYd1ym4fvvb3+rGG2/UsGHDFBsbq9GjR2v27Nnau3evXcM1Ci2BXA+umTPMNAUgPT1dWVlZPscuueQSHTx4UJLkdrslqV1qr66ubvcvAPSu9957Ty+99JK+9a1v2ce4PqHhO9/5ju6++259/etf16hRo5Sfn69vf/vbWr16tSSuU6gYMWKEtm/fruPHj+vQoUMqLi5Wc3OzMjMzuUYhxsn1cLvdampqUm1tbac16ByhKQBXXnml3nrrLZ9j//rXv+yNfVv/MtmyZYt9vqmpSdu3b9eECRP6dKzR7vHHH1daWpquv/56+xjXJzR8/PHHionx/SsoNjbWbjnAdQotycnJSk9PV21trV588UV94Qtf4BqFGCfXIycnR3FxcT41lZWVKisr45o5EcxV6OGquLjY9OvXz/zoRz8yb7/9tnnqqadM//79zfr16+2aNWvWGMuyzLPPPmv27dtnZs2axWO4fczr9Zpzzz3X3HXXXe3OcX2Cb+7cuWbYsGF2y4Fnn33WDBkyxNx55512Ddcp+AoLC81f/vIX8+6775rNmzebSy+91IwZM8Y0NTUZY7hGfa2+vt689tpr5rXXXjOSzIMPPmhee+0189577xljnF2PW265xZxzzjnmpZdeMnv37jVXX301LQccIjQFaOPGjSY7O9skJCSYT33qU+bRRx/1Od/S0mKWL19u3G63SUhIMJ///OfNvn37gjTa6PTiiy8aSeatt95qd47rE3x1dXXm9ttvN+eee65JTEw0F1xwgbnnnntMY2OjXcN1Cr7f//735oILLjDx8fHG7XabW2+91Rw7dsw+zzXqW1u3bjWS2n3NnTvXGOPsepw8edLcdtttJjU11SQlJZkZM2aYgwcPBuHdhB+XMcYEcaILAAAgLLCmCQAAwAFCEwAAgAOEJgAAAAcITQAAAA4QmgAAABwgNAEAADhAaAIAAHCA0AQAAOAAoQkAAMABQhOAqGKM0TXXXKPp06e3O/erX/1KlmXp4MGDQRgZgFBHaAIQVVwulx5//HHt3r1bjzzyiH28oqJCd911l372s5/p3HPP7dHf2dzc3KOvByA4CE0Aos7w4cP1s5/9TEuXLlVFRYWMMbrppps0ZcoUjRkzRtddd50GDBigoUOHKj8/Xx999JH9s4WFhbrqqqv0iU98QoMHD9aMGTO0f/9++/yBAwfkcrn0hz/8QZMmTVJiYqLWr18fjLcJoIexYS+AqPXFL35Rx44d01e+8hX94Ac/0J49e3TFFVdo3rx5+sY3vqGTJ0/qrrvu0qlTp/Tyyy9LkjZs2CCXy6VRo0bpxIkT+v73v68DBw6otLRUMTExOnDggDIzM3X++efrJz/5iS6//HIlJCQoIyMjyO8WQHcRmgBErerqamVnZ+vo0aN65pln9Nprr2n37t168cUX7ZrDhw9r+PDheuutt3TRRRe1e40PP/xQaWlp2rdvn7Kzs+3Q9NOf/lS33357X74dAL2M23MAolZaWppuvvlmXXLJJfrSl76kkpISbd26VQMGDLC/PvWpT0mSfQtu//79mj17ti644AINHDhQmZmZktRu8fgVV1zRt28GQK/rF+wBAEAw9evXT/36nf6rsKWlRXl5ebrvvvva1aWnp0uS8vLyNHz4cD322GPKyMhQS0uLsrOz1dTU5FOfnJzc+4MH0KcITQDwb6NHj9aGDRt0/vnn20HqTEePHtU///lPPfLII/rc5z4nSdqxY0dfDxNAkHB7DgD+7dZbb1VNTY1mzZql4uJivfvuu9q8ebNuvPFGeb1eDRo0SIMHD9ajjz6qd955Ry+//LLuuOOOYA8bQB8hNAHAv2VkZOjvf/+7vF6vpk+fruzsbN1+++2yLEsxMTGKiYlRQUGBSkpKlJ2drW9/+9u6//77gz1sAH2Ep+cAAAAcYKYJAADAAUITAACAA4QmAAAABwhNAAAADhCaAAAAHCA0AQAAOEBoAgAAcIDQBAAA4AChCQAAwAFCEwAAgAOEJgAAAAf+P2VAqTEOp8AkAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from matplotlib import pyplot as plt\n",
    "import numpy as np\n",
    "import matplotlib\n",
    "import random\n",
    "# define our dataset\n",
    "data = np.genfromtxt('taiwan-salary.csv', delimiter=',')\n",
    "years = data[:, 0] # x\n",
    "salarys = data[:, 1] # y\n",
    "m = salarys.size\n",
    "print(\"size\",m)\n",
    "# plot data\n",
    "plt.scatter(years, salarys)\n",
    "plt.xlabel(\"Year\")\n",
    "plt.ylabel(\"Salary(TWD)\")\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "778b4cc9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[ 62.  63.  64.  65.  66.  67.  68.  69.  70.  71.  72.  73.  74.  75.\n",
      "  76.  77.  78.  79.  80.  81.  82.  83.  84.  85.  86.  87.  88.  89.\n",
      "  90.  91.  92.  93.  94.  95.  96.  97.  98.  99. 100. 101. 102. 103.\n",
      " 104. 105.] [ 2254.  3078.  3637.  4266.  5068.  5631.  6727.  9074. 10925. 11682.\n",
      " 12313. 13689. 14265. 15412. 16799. 18707. 21535. 24586. 27097. 29618.\n",
      " 31848. 33783. 35444. 36734. 38521. 39690. 40846. 41856. 41927. 41490.\n",
      " 42023. 42636. 43116. 43448. 44342. 44309. 42036. 44256. 45395. 45483.\n",
      " 45553. 47164. 48362. 48657.]\n"
     ]
    }
   ],
   "source": [
    "# normalization\n",
    "weights = 1\n",
    "train_data = years\n",
    "train_label = salarys\n",
    "print(train_data, train_label)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "668874c4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "[ 62.  63.  64.  65.  66.  67.  68.  69.  70.  71.  72.  73.  74.  75.\n",
      "  76.  77.  78.  79.  80.  81.  82.  83.  84.  85.  86.  87.  88.  89.\n",
      "  90.  91.  92.  93.  94.  95.  96.  97.  98.  99. 100. 101. 102. 103.\n",
      " 104. 105.]\n",
      "422.08887770979646\n",
      "[ 62.  63.  64.  65.  66.  67.  68.  69.  70.  71.  72.  73.  74.  75.\n",
      "  76.  77.  78.  79.  80.  81.  82.  83.  84.  85.  86.  87.  88.  89.\n",
      "  90.  91.  92.  93.  94.  95.  96.  97.  98.  99. 100. 101. 102. 103.\n",
      " 104. 105.]\n",
      "459.7092369704234\n",
      "[ 62.  63.  64.  65.  66.  67.  68.  69.  70.  71.  72.  73.  74.  75.\n",
      "  76.  77.  78.  79.  80.  81.  82.  83.  84.  85.  86.  87.  88.  89.\n",
      "  90.  91.  92.  93.  94.  95.  96.  97.  98.  99. 100. 101. 102. 103.\n",
      " 104. 105.]\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAk0AAAGwCAYAAAC0HlECAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8qNh9FAAAACXBIWXMAAA9hAAAPYQGoP6dpAABUUklEQVR4nO39eXyU5aE+/l+TbRKyDFlIJgMBws5kECEIBFFAIBmFUE/Pr1r5kWK1WEFBKlTl6ClglUU9ak854tLzKa1YsXXPEScJZVEkEEzAMgkgSyAJzCQhyySErDP394+Qh0wWeCbbLLner1der2aeO5N7GMtcPM/9XLdCCCFARERERDfl5ewJEBEREbkDhiYiIiIiGRiaiIiIiGRgaCIiIiKSgaGJiIiISAaGJiIiIiIZGJqIiIiIZPBx9gQ8ic1mw+XLlxEcHAyFQuHs6RAREZEMQghUV1dDo9HAy6vz80kMTT3o8uXLiImJcfY0iIiIqAsKCwsxZMiQTo8zNPWg4OBgAM1/6CEhIU6eDREREclRVVWFmJgY6XO8MwxNPajlklxISAhDExERkZu51dIaLgQnIiIiksGpoWnDhg1QKBR2X2q1WjouhMCGDRug0WgQEBCA2bNnIzc31+456uvrsXLlSkRERCAwMBCLFi1CUVGR3ZiKigqkpKRApVJBpVIhJSUFlZWVdmMKCgqQnJyMwMBAREREYNWqVWhoaOi1105ERETuxelnmuLi4mAymaSvEydOSMdeeeUVvP7669i2bRuOHj0KtVqN+fPno7q6WhqzevVqfPbZZ9i1axcOHjyIq1evYuHChbBardKYxYsX4/jx4zAYDDAYDDh+/DhSUlKk41arFQsWLEBNTQ0OHjyIXbt24ZNPPsGaNWv65g+BiIiIXJ9wovXr14uJEyd2eMxmswm1Wi22bNkiPVZXVydUKpV4++23hRBCVFZWCl9fX7Fr1y5pzKVLl4SXl5cwGAxCCCHy8vIEAHH48GFpTGZmpgAgTp06JYQQYvfu3cLLy0tcunRJGvPhhx8KpVIpLBZLp/Ovq6sTFotF+iosLBQAbvozRERE5FosFousz2+nn2k6c+YMNBoNYmNj8fOf/xznz58HAOTn58NsNiMxMVEaq1QqMWvWLBw6dAgAkJ2djcbGRrsxGo0GOp1OGpOZmQmVSoVp06ZJY6ZPnw6VSmU3RqfTQaPRSGOSkpJQX1+P7OzsTue+efNm6ZKfSqVi3QAREZEHc2pomjZtGv76178iLS0N7733HsxmM2bMmIGysjKYzWYAQFRUlN3PREVFScfMZjP8/PwQGhp60zGRkZHtfndkZKTdmLa/JzQ0FH5+ftKYjqxbtw4Wi0X6KiwsdPBPgIiIiNyFUysH7r33Xul/T5gwAQkJCRg5ciT+8pe/YPr06QDa3/4nhLjlLYFtx3Q0vitj2lIqlVAqlTedCxEREXkGp1+eay0wMBATJkzAmTNnpLvo2p7pKSkpkc4KqdVqNDQ0oKKi4qZjiouL2/2u0tJSuzFtf09FRQUaGxvbnYEiIiKi/smlQlN9fT1OnjyJ6OhoxMbGQq1WIyMjQzre0NCAAwcOYMaMGQCA+Ph4+Pr62o0xmUwwGo3SmISEBFgsFmRlZUljjhw5AovFYjfGaDTCZDJJY9LT06FUKhEfH9+rr5mIiIjcg1Mvz61duxbJyckYOnQoSkpK8NJLL6GqqgpLly6FQqHA6tWrsWnTJowePRqjR4/Gpk2bMGDAACxevBgAoFKp8Oijj2LNmjUIDw9HWFgY1q5diwkTJmDevHkAgPHjx0Ov12PZsmV45513AACPPfYYFi5ciLFjxwIAEhMTodVqkZKSgldffRXl5eVYu3Ytli1bxmZvIiIiJ7PaBLLyy1FSXYfIYH9MjQ2Dt9fNl+r0BqeGpqKiIjz00EO4cuUKBg0ahOnTp+Pw4cMYNmwYAOCZZ55BbW0tVqxYgYqKCkybNg3p6el2e8O88cYb8PHxwQMPPIDa2lrMnTsXO3bsgLe3tzTmgw8+wKpVq6S77BYtWoRt27ZJx729vfHVV19hxYoVuPPOOxEQEIDFixfjtdde66M/CSIiIuqIwWjCxtQ8mCx10mPRKn+sT9ZCr4vu07kohBCiT3+jB6uqqoJKpYLFYuEZKiIiolu41Rkkg9GE5Ttz0DaotIzYvmRyjwQnuZ/f3LCXiIiI+tytziBZbQIbU/PaBSYAEGgOThtT8zBfq+6zS3UutRCciIiIPF/LGaTWgQkAzJY6LN+ZA4PRhKz88nbHWxMATJY6ZOWX9/Jsb+CZJiIiIuozcs4g/ecXRiSMiJD1fCXVnQernsYzTURERNRn5JxBKq1uwJc/XJb1fJHB/j00s1tjaCIiIqI+I/fM0OjIIIT4d35BTIHmNVBTY8N6aGa3xtBEREREfaLJakNpdb2ssS/+RIdX/n+3QYEbd8u1aPl+fbK2T/uauKaJiIiIek19kxXfnb0Cg9GMjLxiVFxrvOl4BQC16kb9wPYlk9vdZad2Uk8TQxMRERH1qJr6Juw7XYK03GLsO1WCq/VN0rGBA3yhjQ7BoXNl7X6uozNIel005mvVbAQnIiIiz1BR04A9J4thMJrxzZlSNFpv3B+nDvFHUlwUknRqTB0eBh9vrw57mjo7g+TtpUDCyPA+ey2dYWgiIiKiLjFb6pCeZ4bBaMaR/HJYbe2LBMID/fC75PG4b4LG7nFXOoMkF0MTERERyXbhSg3Scs0w5JpxrKDyluPLaxrwxAfHsH2JwmXPIMnF0ERERESdEkLglLm6OSgZzThlrrY7Hj8sFPO1Ufjfb8+j9GpD+5+Hc7Y86Q0MTURERGTHZhM4XlSJNGPzGaWLZdekY95eCiSMCEeSTo1EbRSiQvyRea6sw8DUovWWJ+50ZqkthiYiIiJCk9WGrPxyGHLNSMs1o7jqRp+S0scLd40ehKS4KMzXRmHgAD+7n5VbWNmXW570BoYmIiKifqqu0YqDZ64gLdeMjJPFqGzVoRSk9ME94yKh16kxa8wgBCo7jwxytzLpyy1PegNDExERUT9ytb4J+06VwJBrxv5TJahpsErHwgL9MH98FPQ6NWaMCofSx1vWc06NDUO0yh9mS12HG/G2Lqx0ZwxNREREHq78eodSmtGMb89cQYPVJh1r6VDS66Jxx/BQ+Hg7vsOat5cC65O1WL4zBwrALjg5a8uT3sDQRERE5IFMllqk5xZf71AqQ+sKpdiIQOh1aiTFqXHbYBW8eiDM6HXRLrXlSW9gaCIiIvIQ+S0dSkYzjhdW2h2L04QgKU4NvU6N0ZFBUCh6/qyPOxZWOoKhiYiIyE0JIXDSVN18x5vRjNPFNzqUFAogfmiodEYpJmxAt36X1SZkhSF3K6x0BEMTERGRG7HZBI4VVkpnlArKb3Qo+VwPLElxzR1KkSE3v1tNbhDqaJ+4aA+67CaXQgjR0UJ36oKqqiqoVCpYLBaEhIQ4ezpEROQhGls6lIzNHUol1fYdSrPGDIJep8bccVFQDfCVFYbkBiGD0YTlO3Pa3RXX8mzbl0x2++Ak9/ObZ5qIiIhcUF2jFd+euQKD0Yw9J4thqb3RoRSs9ME94yOhj1Nj1thBGOB34+NcThjqLAiZLXVYvjNHCkJWm8DG1LwOawQ8aXsUuRiaiIiIXER1XSP2nS5FmtGMfadLcK1Nh1KiNgpJcZ13KMkJQ/O1atlBKCu/3C58dTTeE7ZHkYuhiYiIyInKrtZjz8nmaoDvzpbZdShpVP5Iur6Q+47hN78LTe5ZoWB/X9lBqL9sjyIXQxMREVEfu1xZi7Tre7xl5ZfbdSiNGBQI/fVqgAmDVbKrAeSeFco8Vybr+VrWQ8nh7tujyMXQRERE1AfOl16VqgF+KLLYHdMNDkGSVo17J6gxKjK4S88v/2yPvPu/WhaQ94ftUeRiaCIiIuoFQgjkXq6Szij9WHxVOqZQAFOGhSIprmc6lAD5Z3sSRkTgk5xLsoJQf9keRS6GJiIiIpludSu/zSaQU1CBr41mfHn8Mkqv3qgG8PFSYMaoCOjj1JivjcKgYGWPzk3uWaHpI8MdCkL9YXsUudjT1IPY00RE5Lk6u5X/hQXjERLgC4PRjPS8YpS26lBqoQrwxfrk8fjp5Jgu/3653UvLd+YA6DgMte5UcrSwUm4RpjuS+/nN0NSDGJqIiDxTZ7fyy9XdIkhHAo4jYz05CDmCockJGJqIiDyP1SYwc+vem96Z5qUAfhYfg4w8M8qvNXY4puXy2MFn73EomHSlkZthyDFyP7+9+nBOREREbuXK1Xps/vrkTQMTANgEMCYqqNPABNj3H8l1q+4loLl7yWqzH9Gyae5Pbh+MhJHhDEw9hAvBiYiIWrlUWYs0oxmGXDO+v2DfoXQzF1ttnHszjhRBspHbtTA0ERFRv3e25KpUDfCvNh1KIyICcf5KzS2fY5jM2gBHiiDZyO1aGJqIiKjfaelQMhjNMBjNOFtq36F0x/Aw6OPUSIyLQrQqADO37r3lrfwpCcPxp4P5PVoEyUZu18LQRERE/YLVJpB9sQIGY/MZpUuVtR2OGxSkxCN3DrdbXC2n18jPx6vHiyDZyO1aePdcD+Ldc0RErsNqEzh09gq+O3cFZ4qv4nhhJcpqGqTjft5edpvjtujsrjS5t/I72n90K450L1HXsHLACRiaiIic71pDE/6w50f8JfMi6hrtQ9EAP2/odWokatXY8KUR5qr2RZRA5/UAcm/l7+lb/ns6iJE9uZ/fvDxHRERuz1LbiL2nimEwmrH3VAkarR2fD7jWYEWiNgqqAN9OAxPQ+V1pLbfy34rccXLpddGYr1Wze8nJGJqIiMgtlVbXIyOvGIZcMw6dvYImGd0ACjT3Gj2jHyfrd7jSXWk9HcTIcQxNRETkNi6W1eBP3+bj0LkrOF9aY7fGZ0xUEHQaFT49dqnTn285g1R+tfOzTK3xrjRqjaGJiIhc2tmSahiMZvz9+0IUlNvf8ebrrcCCCdFYOXc0Rg4KwhfHL900NLUIC/TjXWnkMIYmIiJyKUIInLhkae5QyjXjfGnnxZJNVoEvjl+GXqfGyEFBss8MqVUBPV4PQJ6PoYmIiJzOahP4/kI5DLlmpOcW23Uo+Xor4KVQoL6pfT2AwI11SvO1aod6jby9FNi+ZHK7u9LUvCuNOsHQRERETlHfZMWhc2VIM5qRnleM8lYdSgP8vDFnbCQS46IQpPTBo3/5vtPnaXunmyNnkFzhrrSerieg3sPQREREDn1wd+dD/lpDEw6cLoUh14y9J0tQXd/U4bgQfx8kT4yGXheNL47feo0ScONON70u2qEzSM68K439S+6FoYmIqJ9z5IO7Kx/ylmuN+Of1DqUDP5baXWZTBfjCUtvY7meKq+qxfGcOti+Z3KX911zhDNKttDR9t72MaLbUSa+dwcm1sBG8B7ERnIjcTWcf3B1t0eHI2JLquuYOJaMZmefK7DqUhoYNgF6nxnxtFFb+LeeWrdwHfjsHs17dd8t1Sm3bu12Z1SYwc+teu/DZmju+JnfGRnAiIropq01gY2peh0Gk7QJrXP/fNxv7n18YUVB+Dem5xcguqEDrf5LHhA3A/bdrcK8uGuOjg6FQKJB5rkxWK3f2xQqPu9MtK7+808AEdN5ITs7l5ewJEBGRczjywS1nbGl1AzbtPoXvLzYHJl/vGyGmsPwaPs4uQkF5DRSK5sfltm2XVNdJ65TUKvtLdWqVv1texnLktZPr4JkmIqJ+qjc+uEdHBmHysIH46GhRu/3f2q7VcXStkjusU5KrK+u0yPl4pomIqJ+S+4EcHqjElWp5245sSI7DNz9e6fBYS4TamJoHq01InUqdRR4FmheZt27lbrnT7Se3D0bCyHC3DEwAuvTayfkYmoiIPJjVJpB5rgxfHL+EzHNlsLZakH2rD24ACPD1xqoPj+H3X5286e9p+ZCHArIv+Xl7KbA+WSv9fNvnA9xvrZJc/fm1uzOGJiIiD2UwmjBz61489N5hPLXrOB567zBmbt0Lg9EE4OYf3C1qG60ov9aAgQN8MaOTBcmtP+SvyNwIt22nkqesVXJEf37t7oprmoiIPJDcDqDpI8Lxi4Rh2HW0sN02JaoAX/zkdg30cc3bk/h4e3XY09S6NDLzXJms+blbp1Jv6c+v3R2xp6kHsaeJiFzBrTqAgObG7QlDVDh8vtzukl1EkB8SRoRj6YzhmDw0FF4dfHjfrBG85Xd7UqcSeT72NBER9VO3qgcAgKq6Jnx3tvms0Dh1MPQ6NfQ6NcZGBUuVAJ252bYjLZf8PKlTiagFQxMRkYeRWxGwaKIGT88fg+ERgT36+x3d+43IXTA0ERF5CJtN4IeiSvzzZIms8Q9NHdrjgakF1+qQJ2JoIiJyY01WG7IulCPNaEZabjHMVbc+y9Syrqi3O4BudhmPyB25TOXA5s2boVAosHr1aukxIQQ2bNgAjUaDgIAAzJ49G7m5uXY/V19fj5UrVyIiIgKBgYFYtGgRioqK7MZUVFQgJSUFKpUKKpUKKSkpqKystBtTUFCA5ORkBAYGIiIiAqtWrUJDQ0NvvVwioi6ra7Ri76liPPPxD7jj5T1Y/N4R/CXzIsxVdQj080byRA1+dVcsAHYAEfUklzjTdPToUbz77ru47bbb7B5/5ZVX8Prrr2PHjh0YM2YMXnrpJcyfPx+nT59GcHAwAGD16tVITU3Frl27EB4ejjVr1mDhwoXIzs6Gt7c3AGDx4sUoKiqCwWAAADz22GNISUlBamoqAMBqtWLBggUYNGgQDh48iLKyMixduhRCCPzxj3/swz8JIqKOXa1vwv7TJTAYzdh3qgQ1DVbpWOgAX8zXRkGvU2PGyAj4+zb/3TdlWCjXFRH1IKdXDly9ehWTJ0/GW2+9hZdeegm333473nzzTQghoNFosHr1ajz77LMAms8qRUVFYevWrfj1r38Ni8WCQYMG4f3338eDDz4IALh8+TJiYmKwe/duJCUl4eTJk9BqtTh8+DCmTZsGADh8+DASEhJw6tQpjB07Fl9//TUWLlyIwsJCaDQaAMCuXbvw8MMPo6SkRHZ9ACsHiKgnVdQ0YM/JYqTlmvHNmStoaNOj1EIdosSGRXEdBqGb1QMQUTO5n99Ovzz3xBNPYMGCBZg3b57d4/n5+TCbzUhMTJQeUyqVmDVrFg4dOgQAyM7ORmNjo90YjUYDnU4njcnMzIRKpZICEwBMnz4dKpXKboxOp5MCEwAkJSWhvr4e2dnZnc69vr4eVVVVdl9ERN1RXFWH9zMv4P//p8OY8vIe/Pbjf2HPyRI0NNkwKFjZyc/UY/nOHKnpuzVP2auNyBU49fLcrl27kJ2dje+//77dMbPZDACIioqyezwqKgoXL16Uxvj5+SE0NLTdmJafN5vNiIyMbPf8kZGRdmPa/p7Q0FD4+flJYzqyefNmbNy48VYvk4jopi5cqUFarhmGXDOOFVTaHRsfHQJ9nBrztVF4ZEdWhz8v0LxWaWNqHuZr1QxGRL3EaaGpsLAQTz31FNLT0+Hv3/lO221L1oQQtyxeazumo/FdGdPWunXr8PTTT0vfV1VVISYm5qZzIyISQuCUubo5KBnNOGWutjs+aehA3KtTIylOjWHhzZUAmefKYK7qfF+31hvh8o41ot7htNCUnZ2NkpISxMfHS49ZrVZ888032LZtG06fPg2g+SxQdPSN6/QlJSXSWSG1Wo2GhgZUVFTYnW0qKSnBjBkzpDHFxcXtfn9paand8xw5csTueEVFBRobG9udgWpNqVRCqez4dDkRUWs2m8DxokqkGZvPKF0suyYd8/ZSIGFEOJJ0aiRqoxAV0v4fknILK+WOIyLHOS00zZ07FydOnLB77Je//CXGjRuHZ599FiNGjIBarUZGRgYmTZoEAGhoaMCBAwewdetWAEB8fDx8fX2RkZGBBx54AABgMplgNBrxyiuvAAASEhJgsViQlZWFqVOnAgCOHDkCi8UiBauEhAS8/PLLMJlMUkBLT0+HUqm0C3VERI5otNqQlV+OtFwz0nLNKG51pkjp44W7Rg+CXqfGvPGRGDjA76bP1XqD254YR0SOc1poCg4Ohk6ns3ssMDAQ4eHh0uOrV6/Gpk2bMHr0aIwePRqbNm3CgAEDsHjxYgCASqXCo48+ijVr1iA8PBxhYWFYu3YtJkyYIC0sHz9+PPR6PZYtW4Z33nkHQHPlwMKFCzF27FgAQGJiIrRaLVJSUvDqq6+ivLwca9euxbJly3gXHBE5pK7RioNnrsCQa8aek8WovNYoHQtS+uCecZHQ69SYNWYQApXy/wqeGhuGaJX/LTfC7e3CSqL+zCV6mjrzzDPPoLa2FitWrEBFRQWmTZuG9PR0qaMJAN544w34+PjggQceQG1tLebOnYsdO3ZIHU0A8MEHH2DVqlXSXXaLFi3Ctm3bpOPe3t746quvsGLFCtx5550ICAjA4sWL8dprr/XdiyUit1Vd14h9p0uRZjRj3+kSXGvVoRQW6If546OQpIvCnaMioPTx7vR5blYPwI1wiZzP6T1NnoQ9TUT9R3lNA/bkFcOQa8bBM1fQYL3RoRSt8kdSnBp6nRpThoXCx/vW7S4Go6ldEWV0B0WUcscRkXxyP78ZmnoQQxORZzNZapGeWwyD0Ywj+WWwtfrbc0REIPTX73i7bYjqlnf5tmYwmrB8Z067y24tz7B9yWS7QMTCSqKeJffz26UvzxEROVv+lRoYrt/x9kNhpd2xOE0IkuLUuFenxqjIIIeCUgurTWBjal6H65Q661/iRrhEzsHQRETUihACJ03VMOSakWY043TxjQ4lhQKIHxoqnVGKCRvQ7d+XlV9ud6mt3XzA/iUiV8HQRET9ns0mcKywUiqbLCi/0aHkc/2sTlJcc4dSZAcdSt3B/iUi98HQRET9UqPVhiPny2HINSE9txgl1fYdSrPGNHcozR0XBdUA316bB/uXiNwHQxMR9Rt1jVZ882MpDLlm/PNkCSy1NzqUgpU+uGd8JPRxaswaOwgD/Prmr0f2LxG5D4YmIvJo1XWN2HuqBGm5Zuw7VYraxhsdSuGBfkiMi0JSnBoJI8Nv2qHUW9i/ROQ+GJqIyOOUXa1HRl4x0nLN+O5smV2HkkbljySdGvo4NaYMd41b9fW6aGxfMrld/5Ka/UtELoWhiYg8wuXKWmkh99EL5fYdSoMCce/1O94mDHasQ6mv6HXRmK9Vs3+JyIUxNBGR2zpXehUGoxnpuWb8UGSxOzZhsApJcVHQ69QYFRncyTO4FvYvEbk2hiYichtCCORerpLOKJ0puSodUyiAO4aFIUnXXA3QEx1KREStMTQRkUuz2QRyCiqkVu6iilrpmK+3AgkjI6CPU2O+NgqDgpVOnGnHuOUJkedgaCIil9NotSHzXBkMuWZk5BWjtFWHkr+vF2aPiYRep8accZFQBfReh1J3cXNdIs/C0ERELqG2wYpvzpQizWjGnpPFqKprko4F+/tg3vjmaoBZYwYhwK/vqwEc1dkmvGZLHZbvzGm3CS8RuT6GJiJyGkttI/adKoHBaMaBH+07lCKC/DBfq4Zep0bCiHD4+Xg5caaO6comvETk+hiaiKhPXbneoWQwmnHo3BU0Wm9Ei8EDA6DXNQelyUND3TZQcBNeIs/E0EREva6o4hrScpvLJo9eKIdodQpmdGQQkuKag1KcJsQlO5QcxU14iTwTQxMR9YqzJVelaoATl+w7lG4bokJSXHPZ5KjIICfNsPdwE14iz8TQREQ9oqVDqaUa4GyrDiUvBTBleJh0RmnwwAAnzrT3cRNeIs/E0EREXWa1CWRfrJDOKF2qtO9QunNUc4fSPG0UIoJcr0Opt3ATXiLPxNBERA5paLIh83wZDEYzMvLMuHK1QToW4OuN2WMHSR1KIf6u26HU27gJL5HnYWgioluqbbDiwI+lSMtt7lCqbtWhFOLvg7njm/d4u3u0e3Qo9QQ5Td/chJfIszA0EVGHLLWN2HuqWOpQqmu0SccigpRIimsum5zuZh1KPcGRpm9uwkvkORiaiEhSUl2HjLxipOUW49DZK2iy3ViNMyQ0APrrC7knuXGHUnex6Zuo/2JoIurnCsuvIS3XjLRcM76/WGHXoTQmKgj6ODWSdGpooz2jQ6k72PRN1L8xNBH1M0IInC25KlUD5F6usjs+MWagdOlt5CDP61DqDjZ9E/VvDE1E/YAQAicuWaSgdL60RjrmpWjuFdLHqZEYp4bGwzuUuoNN30T9G0MTkYey2gSOXiiHwWhGeq4Zl1udIfHz9sLM0RFIiovCvPFRCO9HHUrdwaZvov6NoYnIg9Q3WXHoXBnSjGZk5BWjrOZGh9IAP2/MGRuJxLgo3DMuEsH9uEOpq9j0TdS/MTQRublrDU04cLoUhlwz9p4sQXX9jQ4lVYAv5l3vULprdAT8fftHh1JvYdM3Uf/G0ETkhizXGrHnZDHScps7lOqbbnQoRQYrpT3epsaGwde7f3Uo9TY2fRP1XwxNRG6ipKoO6XnNQSnzXJldh1JMWADu1UUjKU6NSTED4cUzHb2KTd9E/RNDE5ELKyhr7lAy5JqRU2DfoTQ2KhhJOjX0cWqMjw7u9x1KfY1N30T9D0MTkQsRQuBMS4eS0Yw8k32H0u0xA6HXqZEUp0ZsRKCTZum55OwnR0T9F0MTkZMJIfBDUXOHUlquGflXbnQoeXspMC02DElxaiTGRSFaxQ6l3uLIfnJE1D8xNBE5QZPVhqwL5UjPbV6jZGrToXTX6Agk6dSYNz4KYYF+Tpxp/8D95IhIDoYmoj5S32TFd2evwHC9Q6niWqN0LNDPG3PGRSIpTo054yIRpOT/NfsK95MjIrn4NzNRL6qpb8L+6x1K+06V4GqrDqWBA3wx/3qH0p2j2KHkLNxPjojkYmgi6mEVNQ3XO5SK8c2ZUjS06lBSh/gjMa45KE0dHgYfdig5HfeTIyK5GJqIekBxVR3Sr1cDHD5fDmurDqXh4QOkaoCJQ9ih5Gq4nxwRycXQRNRFF8tqmjuUjGbkFFTaHRunDoZe19zKPTaKHUqujPvJEZFcDE1EMgkhcLq4WupQOmWutjs+aehA6OOaO5SGs0PJbXA/OSKSi6GJ6CZsNoEfiiphyDUjzWjGhbJr0jFvLwWmjwiDPk6N+Vo11CpevnFX3E+OiORgaCJqo8lqQ1Z+OQy5ZqTnFsNc1apDyccLd48eBL1OjbnjIhHKDiWPwf3kiOhWGJqIANQ13uhQ2nPSvkMpSOmDOeMioY9TY/bYQQhkh5LH4n5yRHQz/Nuf+q2r9U3Yf7oEBmNzh1JNg1U6FjrAF4laNZJ0UZgxkh1KRETE0ET9TEVNAzJOFiM914xvzlxp16HUshnuHcND2aFERER2GJrI45ktdUjPa77j7Uh++w4lvS4aep0atw1WsUPJQ1ltgmuViKjbGJrII124UtN8x1uuGcfadCiNjw5BUlwU7tVFY0xUEDuUPJzBaGp3V1w074ojoi5gaCKPIITAKXNzh1JabvsOpfhhoVKH0tDwAU6aJfU1g9GE5Ttz2pVWmi11WL4zB9uXTGZwIiLZGJrIbdlsAscKK5F2/YzSxVYdSj7X74JKjFMjURuFqBB2KPU3VpvAxtS8Dlu+BZqLKzem5mG+Vs1LdUQkC0MTuZXGlg6l62eUSqrrpWNKHy/cPWYQ9HFqzB0fiYED2KHUn2Xll9tdkmtLADBZ6pCVX86aASKShaGJXF5doxXfnrnRoWSpvdGhFKz0wT3jmzuUZo0dhAF+/E+ampVUdx6YujKOiIifMOSSqusase90KdKMZuw7XYJrrTqUwgL9kKiNQpJOjRkjw6H0YYcStRcZLO+SrNxxREQMTeQyyq7WY8/JYhiMZnx3tgwN1hsdShqVP5KkDiXeLk63NjU2DNEqf5gtdR2ua1KgeW+5qbFhfT01InJTDE3kVCZLLdKMZhhyzcjKL0erCiWMiAiUyiZvG6JiNQA5xNtLgfXJWizfmQMFYBecWv5LWp+sZQAnItkYmqjPnS+9irTcYhhyzfihsNLumG5wCJK0auh1aoyKZIcSdY9eF43tSya362lSs6eJiLqAoYl6nRACeaYq6YzSj8VXpWMKBTBlWCiSrncoxYSxQ4l6ll4XjflaNRvBiajbHA5NQggcOHAA3377LS5cuIBr165h0KBBmDRpEubNm4eYmJjemCe5meYOpQoYrgelwvJa6VhLh5Jep8Z8bRQX4lKv877+3xwRUXcohBAdrZFsp7a2Fm+88QbeeustlJWVYeLEiRg8eDACAgJQXl4Oo9GIy5cvIzExEb/73e8wffr03p67y6mqqoJKpYLFYkFISIizp9PnGq02HD5fBoPRjPS8YpS26lDy9/XCrDGDkBSnxtxxUVAN8HXiTImIiG6Q+/kt+0zTmDFjMG3aNLz99ttISkqCr2/7D72LFy/ib3/7Gx588EG88MILWLZsWddmT26jrtGKb34shSHXjD15xaiqa5KOBSt9MHd8JPQ6Ne4eww4lIiJyb15yB3799df4+OOPsXDhwg4DEwAMGzYM69atw5kzZzB79uxbPuf27dtx2223ISQkBCEhIUhISMDXX38tHRdCYMOGDdBoNAgICMDs2bORm5tr9xz19fVYuXIlIiIiEBgYiEWLFqGoqMhuTEVFBVJSUqBSqaBSqZCSkoLKykq7MQUFBUhOTkZgYCAiIiKwatUqNDQ0yPvD6Weq6hrxxfFLWPFBNia9mIHH3s/GpzmXUFXXhIggPzw0dSh2/PIOZP/nfLz580nQ66IZmIiIyO3J/iTT6XSyn9TPzw+jR4++5bghQ4Zgy5YtGDVqFADgL3/5C37yk5/g2LFjiIuLwyuvvILXX38dO3bswJgxY/DSSy9h/vz5OH36NIKDgwEAq1evRmpqKnbt2oXw8HCsWbMGCxcuRHZ2Nry9m0sPFy9ejKKiIhgMBgDAY489hpSUFKSmpgIArFYrFixYgEGDBuHgwYMoKyvD0qVLIYTAH//4R9mv25OVXa1HRl7zHW/fnb2CRuuNq7qDBwYgKa75jrf4YaFcYEtERB5J9pqm1s6cOYMvvvgCFy5cgEKhQGxsLO6//36MGDGi2xMKCwvDq6++ikceeQQajQarV6/Gs88+C6D5rFJUVBS2bt2KX//617BYLBg0aBDef/99PPjggwCAy5cvIyYmBrt370ZSUhJOnjwJrVaLw4cPY9q0aQCAw4cPIyEhAadOncLYsWPx9ddfY+HChSgsLIRGowEA7Nq1Cw8//DBKSkpkr0/ytDVNlytrkZZrhsFoxtEL9h1KIwc1dyjp46KhGxzCagDqc1ab4B1xRNQjenxNU4vNmzfjd7/7HWw2GyIjIyGEQGlpKZ577jls2rQJa9eu7dKErVYr/vGPf6CmpgYJCQnIz8+H2WxGYmKiNEapVGLWrFk4dOgQfv3rXyM7OxuNjY12YzQaDXQ6HQ4dOoSkpCRkZmZCpVJJgQkApk+fDpVKhUOHDmHs2LHIzMyETqeTAhMAJCUlob6+HtnZ2ZgzZ06Hc66vr0d9/Y3FzlVVVV167a7kXOlVaTPcfxVZ7I5NGKy6XjYZhVGRwU6aIRFgMJradS9Fs3uJiHqZQ6Fp3759eOGFF/Cf//mfeOqppxAaGgoAKC8vx5tvvonnnnsOU6dOxd133y37OU+cOIGEhATU1dUhKCgIn332GbRaLQ4dOgQAiIqKshsfFRWFixcvAgDMZjP8/PykebQeYzabpTGRkZHtfm9kZKTdmLa/JzQ0FH5+ftKYjmzevBkbN26U/VpdkRACuZerpDNKZ0rsO5TuGBaGxLgodiiRyzAYTVi+M6fd1ihmSx2W78zB9iWTGZyIqFc4FJrefvtt/OpXv8KGDRvsHg8LC8OLL74Is9mM7du3OxSaxo4di+PHj6OyshKffPIJli5digMHDkjH2172EULc8lJQ2zEdje/KmLbWrVuHp59+Wvq+qqrKLXqqrDaBnILrHUpGMy5V3uhQ8vVWYMbICCTFNXcoDQpWOnGmRPasNoGNqXkd7iUn0Lw9ysbUPMzXqnmpjoh6nEOhKSsrC++//36nx1NSUvCLX/zCoQn4+flJC8GnTJmCo0eP4g9/+IO0jslsNiM6+sa/GktKSqSzQmq1Gg0NDaioqLA721RSUoIZM2ZIY4qLi9v93tLSUrvnOXLkiN3xiooKNDY2tjsD1ZpSqYRS6R6hoqHJhszzZUjLNSM9txhXrtp3KM0e01wNMGdcJFQB7FAi15SVX253Sa4tAcBkqUNWfjnLLImoxzkUmoqLizF8+PBOj8fGxt70cpYcQgjU19cjNjYWarUaGRkZmDRpEgCgoaEBBw4cwNatWwEA8fHx8PX1RUZGBh544AEAgMlkgtFoxCuvvAIASEhIgMViQVZWFqZOnQoAOHLkCCwWixSsEhIS8PLLL8NkMkkBLT09HUqlEvHx8d16Pc5U22DFgR9LkZZrxp6Txahu3aHk74N545svu80aMwgBft5OnCmRPCXVnQemrowjInKEQ6Gprq4Ofn5+nR739fV1qNvoP/7jP3DvvfciJiYG1dXV2LVrF/bv3w+DwQCFQoHVq1dj06ZNGD16NEaPHo1NmzZhwIABWLx4MQBApVLh0UcfxZo1axAeHo6wsDCsXbsWEyZMwLx58wAA48ePh16vx7Jly/DOO+8AaK4cWLhwIcaOHQsASExMhFarRUpKCl599VWUl5dj7dq1WLZsmdvdBWepbcS+UyUwGM3Y/2MJ6hpt0rGIICUS46Kgj1Nj+ohw+PnIrukicglyt9zh1jxE1BscvnvuT3/6E4KCgjo8Vl1d7dBzFRcXIyUlBSaTCSqVCrfddhsMBgPmz58PAHjmmWdQW1uLFStWoKKiAtOmTUN6errU0QQAb7zxBnx8fPDAAw+gtrYWc+fOxY4dO6SOJgD44IMPsGrVKukuu0WLFmHbtm3ScW9vb3z11VdYsWIF7rzzTgQEBGDx4sV47bXXHHo9zlJafaNDKfOcfYfSkNAA6K93KE0ayg4lcm9TY8MQrfKH2VLX4bomBQC1qrl+gIiopznU0zR8+HBZfTz5+fndmpS76suepqKKa0jLLUaa0YyjF8vR+l0cHRl0vRpAjTgNO5TIs7TcPQfALji1/FfOu+eIyFG90tN04cKF7s6LuuFsSTXScothMJpx4pJ9h9LEISokXQ9KIwd1fCaQyBPoddHYvmRyu54mNXuaiKiXORSazp49K93pRn2jocmGP/zzRxiMZpwrrZEe91IAdwwPg16nRmKcGoMHBjhxlkR9S6+Lxnytmo3gRNSnHApNY8aMweDBgzFnzhzp62Z301H3+XorsPuEGflXauDrrcCdoyKgj1NjnjYKEUHuUXdA1Bu8vRSsFSCiPuVQaDpw4AAOHDiA/fv348knn0RdXR2GDh2Ke+65RwpRgwcP7q259ksKhQKr5zVvfjxnXCRC/NmhRERE5Axd2rAXABobG5GZmYn9+/dj//79OHz4MOrr6zFq1CicPn26p+fpFjxtw14iIqL+QO7nd5dDU4va2locPHgQaWlpeO+993D16lVYrdbuPKXbYmgiIiJyP71y9xzQXHB56NAh7Nu3D/v378fRo0cRGxuLWbNmYfv27Zg1a1a3Jk5ERETkihwKTbNmzcLRo0cxcuRI3H333Vi5ciVmzZp10/3ZiIiIiDyBQ6Hp0KFDiI6Oxpw5czB79mzcfffdiIiI6K25EREREbkMhzYfq6ysxLvvvosBAwZg69atGDx4MCZMmIAnn3wSH3/8MUpLS3trnkTUT1htApnnyvDF8UvIPFcGq61byy6JiHpMtxaCV1dX4+DBg9L6ph9++AGjR4+G0WjsyTm6DS4EJ+oeg9HUruk7mk3fRNTL5H5+d2ub+8DAQISFhSEsLAyhoaHw8fHByZMnu/OURNRPtewp1zowAYDZUoflO3NgMJqcNDMiomYOrWmy2Wz4/vvvsX//fuzbtw/fffcdampqpJbw//mf/8GcOXN6a65E5KGsNoGNqXno6LS3QPNmvBtT8zBfq+ZWKUTkNA6FpoEDB6KmpgbR0dGYPXs2Xn/9dcyZMwcjR47srfkRUT+QlV/e7gxTawKAyVKHrPxybp1CRE7jUGh69dVXMWfOHIwZM6a35kNE/VBJdeeBqSvjiIh6g0NrmlasWIGBAwf20lSIqL+KDPbv0XFERL3BodDUzR1XiIg6NDU2DNEqf3S2WkmB5rvopsaG9eW0iIjsdOvuOSKinuDtpcD6ZC0AtAtOLd+vT9ZyETgROZXDe8+lpaVBpVLddMyiRYu6PCEi6p/0umhsXzK5XU+Tmj1NROQiHCq39PK69YkphUIBq9XarUm5K5ZbEnWf1SaQlV+Okuo6RAY3X5LjGSYi6k1yP78dPtNkNpsRGRnZrckREXXG20vBWgEickkOrWlSKPivPSIiIuqfePccERERkQwOhaalS5ciICCgt+ZCRERE5LJkr2mqqanBn//8Z9lPXFNTg8DAwC5NioiIiMjVyD7TNGrUKGzatAmXL1/udIwQAhkZGbj33nvx3//93z0yQSIiIiJXIPtM0/79+/HCCy9g48aNuP322zFlyhRoNBr4+/ujoqICeXl5yMzMhK+vL9atW4fHHnusN+dNRERE1Kcc6mkCgKKiIvzjH//AN998gwsXLqC2thYRERGYNGkSkpKScN9998nqc/JE7Gki6hi7l4jIlcn9/HY4NFHnGJqI2jMYTe1avqPZ8k1ELkTu53eXTgnt37+/q/Mion7EYDRh+c4cu8AEAGZLHZbvzIHBaHLSzIiIHNel0KTX6zFy5Ei89NJLKCws7Ok5EZEHsNoENqbmoaNT2S2PbUzNg9XGk91E5B66FJouX76Mp556Cp9++iliY2ORlJSEv//972hoaOjp+RGRm8rKL293hqk1AcBkqUNWfnnfTYqIqBu6FJrCwsKwatUq5OTk4Pvvv8fYsWPxxBNPIDo6GqtWrcIPP/zQ0/MkIjdTUt15YOrKOCIiZ+v2bW633347nnvuOTzxxBOoqanB//t//w/x8fG46667kJub2xNzJCI3FBns36PjiIicrcuhqbGxER9//DHuu+8+DBs2DGlpadi2bRuKi4uRn5+PmJgY/OxnP+vJuRKRG5kaG4ZolT86KxZQoPkuuqmxYX05LSKiLutSaFq5ciWio6Px+OOPY8yYMTh27BgyMzPxq1/9CoGBgYiJicGWLVtw6tSpnp4vEbkJby8F1idrAaBdcGr5fn2yln1NROQ2ZDeCt5aXl4c//vGP+Pd//3f4+fl1OEaj0WDfvn3dmhwRuTe9Lhrbl0xu19OkZk8TEbkhh0NTY2Mjhg4dimnTpnUamADAx8cHs2bN6tbkiMj96XXRmK9VsxGciNyew5fnfH198dlnn/XGXIjIQ3l7KZAwMhw/uX0wEkaGMzARkVvq0pqmf/u3f8Pnn3/ew1MhIiIicl1dWtM0atQo/P73v8ehQ4cQHx+PwMBAu+OrVq3qkckRERERuYoubdgbGxvb+RMqFDh//ny3JuWuuGEvERGR+5H7+d2lM035+fldnhgRERGRO+p2IzgRERFRf9ClM00AUFRUhC+//BIFBQXtNup9/fXXuz0xIiIiIlfSpdD0z3/+E4sWLUJsbCxOnz4NnU6HCxcuQAiByZMn9/QciYiIiJyuS5fn1q1bhzVr1sBoNMLf3x+ffPIJCgsLMWvWLO43R0RERB6pS6Hp5MmTWLp0KYDm5u/a2loEBQXhxRdfxNatW3t0gkRERESuoEuhKTAwEPX19QCa95g7d+6cdOzKlSs9MzMiIiIiF9KlNU3Tp0/Hd999B61WiwULFmDNmjU4ceIEPv30U0yfPr2n50hELspqE9xTjoj6jS6Fptdffx1Xr14FAGzYsAFXr17FRx99hFGjRuGNN97o0QkSkWsyGE3YmJoHk6VOeixa5Y/1yVroddFOnBkRUe/oUiM4dYyN4NRfGIwmLN+Zg7Z/ebScY9q+ZDKDExG5Dbmf3yy3JCKHWG0CG1Pz2gUmANJjG1PzYLXx32NE5FlkX54LDQ2FQiFvrUJ5eXmXJ0REri0rv9zuklxbAoDJUoes/HIkjAzvu4kREfUy2aHpzTff7MVpEJG7KKnuPDB1ZRwRkbuQHZpaepmIqH+LDPbv0XFERO6iy3vPtaitrUVjY6PdY1wETeS5psaGIVrlD7OlrsN1TQoAalVz/QARkSfp0kLwmpoaPPnkk4iMjERQUBBCQ0PtvojIc3l7KbA+WQvgxt1yLVq+X5+sZV8TEXmcLoWmZ555Bnv37sVbb70FpVKJP/3pT9i4cSM0Gg3++te/9vQcicjF6HXR2L5kMtQq+0twapU/6waIyGN1qadp6NCh+Otf/4rZs2cjJCQEOTk5GDVqFN5//318+OGH2L17d2/M1eWxp4n6GzaCE5EnkPv53aU1TeXl5YiNjQXQvH6ppWJg5syZWL58eVeekojckLeXgrUCRNRvdOny3IgRI3DhwgUAgFarxd///ncAQGpqKgYOHNhTcyMiIiJyGV0KTb/85S/xww8/AADWrVsnrW36zW9+g9/+9reyn2fz5s244447EBwcjMjISNx///04ffq03RghBDZs2ACNRoOAgADMnj0bubm5dmPq6+uxcuVKREREIDAwEIsWLUJRUZHdmIqKCqSkpEClUkGlUiElJQWVlZV2YwoKCpCcnIzAwEBERERg1apVaGhocOBPhoiIiDxVj+w9V1BQgO+//x4jR47ExIkTZf+cXq/Hz3/+c9xxxx1oamrC888/jxMnTiAvLw+BgYEAgK1bt+Lll1/Gjh07MGbMGLz00kv45ptvcPr0aQQHBwMAli9fjtTUVOzYsQPh4eFYs2YNysvLkZ2dDW9vbwDAvffei6KiIrz77rsAgMceewzDhw9HamoqAMBqteL222/HoEGD8F//9V8oKyvD0qVL8dOf/hR//OMfZb0ermkiIiJyP7I/v4UDDh8+LHbv3m332F/+8hcxfPhwMWjQILFs2TJRV1fnyFPaKSkpEQDEgQMHhBBC2Gw2oVarxZYtW6QxdXV1QqVSibffflsIIURlZaXw9fUVu3btksZcunRJeHl5CYPBIIQQIi8vTwAQhw8flsZkZmYKAOLUqVNCCCF2794tvLy8xKVLl6QxH374oVAqlcJisXQ437q6OmGxWKSvwsJCAaDT8UREROR6LBaLrM9vhy7PbdiwAf/617+k70+cOIFHH30U8+bNw7p165CamorNmzd3JeQBACwWCwAgLKy5FC8/Px9msxmJiYnSGKVSiVmzZuHQoUMAgOzsbDQ2NtqN0Wg00Ol00pjMzEyoVCpMmzZNGjN9+nSoVCq7MTqdDhqNRhqTlJSE+vp6ZGdndzjfzZs3S5f7VCoVYmJiuvzaiYiIyLU5FJqOHz+OuXPnSt/v2rUL06ZNw3vvvYff/OY3+O///m9pUbijhBB4+umnMXPmTOh0OgCA2WwGAERFRdmNjYqKko6ZzWb4+fm1K9VsOyYyMrLd74yMjLQb0/b3hIaGws/PTxrT1rp162CxWKSvwsJCR182ERERuQmHKgcqKirsgsWBAweg1+ul7++4444uB4cnn3wS//rXv3Dw4MF2xxQK+94XIUS7x9pqO6aj8V0Z05pSqYRSqbzpPIiIiMgzOHSmKSoqCvn5+QCAhoYG5OTkICEhQTpeXV0NX19fhyexcuVKfPnll9i3bx+GDBkiPa5WqwGg3ZmekpISKbyp1Wo0NDSgoqLipmOKi4vb/d7S0lK7MW1/T0VFBRobG9udgSIiIqL+x6HQpNfr8dxzz+Hbb7/FunXrMGDAANx1113S8X/9618YOXKk7OcTQuDJJ5/Ep59+ir1790qFmS1iY2OhVquRkZEhPdbQ0IADBw5gxowZAID4+Hj4+vrajTGZTDAajdKYhIQEWCwWZGVlSWOOHDkCi8ViN8ZoNMJkMklj0tPToVQqER8fL/s1ERERkYdyZHV5SUmJmDlzplAoFCI4OFh8+umndsfvuece8R//8R+yn2/58uVCpVKJ/fv3C5PJJH1du3ZNGrNlyxahUqnEp59+Kk6cOCEeeughER0dLaqqqqQxjz/+uBgyZIjYs2ePyMnJEffcc4+YOHGiaGpqksbo9Xpx2223iczMTJGZmSkmTJggFi5cKB1vamoSOp1OzJ07V+Tk5Ig9e/aIIUOGiCeffFL265G7+p7IlTVZbeLQ2Svi82NF4tDZK6LJanP2lIiIepXcz+8u9TRZLBYEBQVJHUgtysvLERQUBD8/P1nP09laoT//+c94+OGHW0IdNm7ciHfeeQcVFRWYNm0a/ud//kdaLA4AdXV1+O1vf4u//e1vqK2txdy5c/HWW2/Z3c1WXl6OVatW4csvvwQALFq0CNu2bbNrMC8oKMCKFSuwd+9eBAQEYPHixXjttddkr1tiTxO5O4PRhI2peTBZ6qTHolX+WJ+s5Sa8ROSx5H5+90i5JTVjaCJ3ZjCasHxnDtr+hdDyT5vtSyYzOBGRR5L7+d2lbVSIyLNYbQIbU/PaBSYA0mMbU/NgtfHfWETUfzE0ERGy8svtLsm1JQCYLHXIyi/vu0kREbkYhiYiQkl154GpK+OIiDwRQxMRITLYv0fHERF5IoYmIsLU2DBEq/zRWc++As130U2NDevLaRERuRSGJiKCt5cC65O1ANAuOLV8vz5ZC2+vm29fRETkyRiaiAgAoNdFY/uSyVCr7C/BqVX+rBsgIoKDG/YSkWfT66IxX6tGVn45SqrrEBncfEmOZ5iIiBiaiKgNby8FEkaGO3saREQuh5fniIiIiGRgaCIiIiKSgZfniPoJq01wrRIRUTcwNBH1AwajCRtT8+y2SolW+WN9spZ3xRERycTLc0QezmA0YfnOnHZ7y5ktdVi+MwcGo8lJMyMici8MTUQezGoT2JiaB9HBsZbHNqbmwWrraAQREbXG0ETkwbLyy9udYWpNADBZ6pCVX953kyIiclMMTUQerKS688DUlXFERP0ZQxORB4sM9r/1IAfGERH1ZwxNRB5samwYolX+7TbhbaFA8110U2PD+nJaRERuiaGJyIN5eymwPlkLAO2CU8v365O17GsiIpKBoYnIw+l10di+ZDLUKvtLcGqVP7YvmcyeJiIimVhuSeTm5DR963XRmK9VsxGciKgbGJqI3JgjTd/eXgokjAzv6ykSEXkMXp4jclNs+iYi6lsMTURuiE3fRER9j6GJyA2x6ZuIqO8xNBG5ITZ9ExH1PYYmIjfEpm8ior7H0ETkhtj0TUTU9xiaiFyQ1SaQea4MXxy/hMxzZe0WdLPpm4io77GnicjFyO1eamn6bjtW3UlPExERdY9CCMF7kntIVVUVVCoVLBYLQkJCnD0dckMt3Utt/0/Zcr6oo21P5DSCExFR5+R+fvNME5GLuFX3kgLN3UvztWq7UMSmbyKivsE1TUR96GZrldi9RETk2nimiaiP3GqtEruXiIhcG880EfUBOfvEsXuJiMi1MTQR9TK5+8TFDwtl9xIRkQtjaCLqZXLXKmVfrGD3EhGRC2NoIupljqxVauleUqvsL8GpVf4d1g0QEVHf4UJwol7m6FolvS4a87Vqdi8REbkYhiaiXtayT5zZUtfhuiYFms8ktV6rxO4lIiLXw8tzRL2M+8QREXkGhiaiPsC1SkRE7o+X54i6Se7eb1yrRETk3hiaiLrhVi3fbXGtEhGR++LlOaIuktPyTUREnoOhiagL5LZ8t96Ql4iI3BtDE1EXyG35zsov77tJERFRr+KaJqJO3GyBtyMt30RE5BkYmog6cKsF3o62fBMRkfvj5TmiNuQs8G5p+e6sLECB5pDVuuWbiIjcG0MTUStyF3gDYMs3EVE/w9BE1IojC7zZ8k1E1L9wTRP1K7dq73Z0gTdbvomI+g+GJuo35LR3d2WBN1u+iYj6B16eo35Bbns3F3gTEVFnGJrI4znS3u3tpeACbyIi6hBDE3k8R9u7ucCbiIg6wjVN5PG60t7NBd5ERNQWQxN5vK62d3OBNxERtcbLc+TxuLibiIh6glND0zfffIPk5GRoNBooFAp8/vnndseFENiwYQM0Gg0CAgIwe/Zs5Obm2o2pr6/HypUrERERgcDAQCxatAhFRUV2YyoqKpCSkgKVSgWVSoWUlBRUVlbajSkoKEBycjICAwMRERGBVatWoaGhoTdeNvUxLu4mIqKe4NTQVFNTg4kTJ2Lbtm0dHn/llVfw+uuvY9u2bTh69CjUajXmz5+P6upqaczq1avx2WefYdeuXTh48CCuXr2KhQsXwmq1SmMWL16M48ePw2AwwGAw4Pjx40hJSZGOW61WLFiwADU1NTh48CB27dqFTz75BGvWrOm9F099iou7iYio24SLACA+++wz6XubzSbUarXYsmWL9FhdXZ1QqVTi7bffFkIIUVlZKXx9fcWuXbukMZcuXRJeXl7CYDAIIYTIy8sTAMThw4elMZmZmQKAOHXqlBBCiN27dwsvLy9x6dIlacyHH34olEqlsFgssl+DxWIRABz6GepbTVabOHT2ivj8WJE4dPaKaLLanD0lIiJyMrmf3y67pik/Px9msxmJiYnSY0qlErNmzcKhQ4cAANnZ2WhsbLQbo9FooNPppDGZmZlQqVSYNm2aNGb69OlQqVR2Y3Q6HTQajTQmKSkJ9fX1yM7O7nSO9fX1qKqqsvsi19ayuPsntw9GwshwXpIjIiLZXDY0mc1mAEBUVJTd41FRUdIxs9kMPz8/hIaG3nRMZGRku+ePjIy0G9P294SGhsLPz08a05HNmzdL66RUKhViYmIcfJVERETkLlw2NLVQKOzPBAgh2j3WVtsxHY3vypi21q1bB4vFIn0VFhbedF5ERETkvlw2NKnVagBod6anpKREOiukVqvR0NCAioqKm44pLi5u9/ylpaV2Y9r+noqKCjQ2NrY7A9WaUqlESEiI3Rc5h9UmkHmuDF8cv4TMc2Ww2jraNIWIiKjrXDY0xcbGQq1WIyMjQ3qsoaEBBw4cwIwZMwAA8fHx8PX1tRtjMplgNBqlMQkJCbBYLMjKypLGHDlyBBaLxW6M0WiEyWSSxqSnp0OpVCI+Pr5XXyd1n8Fowsyte/HQe4fx1K7jeOi9w5i5da+0CS8REVFPcGoj+NWrV3H27Fnp+/z8fBw/fhxhYWEYOnQoVq9ejU2bNmH06NEYPXo0Nm3ahAEDBmDx4sUAAJVKhUcffRRr1qxBeHg4wsLCsHbtWkyYMAHz5s0DAIwfPx56vR7Lli3DO++8AwB47LHHsHDhQowdOxYAkJiYCK1Wi5SUFLz66qsoLy/H2rVrsWzZMp49cnEGownLd+a024zXbKnD8p05rBMgIqIe49TQ9P3332POnDnS908//TQAYOnSpdixYweeeeYZ1NbWYsWKFaioqMC0adOQnp6O4OBg6WfeeOMN+Pj44IEHHkBtbS3mzp2LHTt2wNvbWxrzwQcfYNWqVdJddosWLbLrhvL29sZXX32FFStW4M4770RAQAAWL16M1157rbf/CKgbrDaBjal57QIT0LwJrwLAxtQ8zNeqeZccERF1m0IIwcUfPaSqqgoqlQoWi4VnqPpA5rkyPPTe4VuO+3DZdO4hR0REnZL7+e2ya5qIbqWkuq5HxxEREd0MQxO5rchg/1sPcmAcERHRzTA0kduaGhuGaJV/u014WygARKv8MTU2rC+nRUREHoqhidyWt5cC65O1ANAuOLV8vz5Zy0XgRETUIxiayK3pddHYvmQy1Cr7S3BqlT/rBoiIqEc5tXKAqCfoddGYr1UjK78cJdV1iAxuviTHM0xERNSTGJrII3h7KVgrQEREvYqX54iIiIhkYGgiIiIikoGhiYiIiEgGhiYiIiIiGRiaiIiIiGTg3XPksqw2wRoBIiJyGQxN5JIMRhM2pubBZLmx2W60yh/rk7UsrCQiIqfg5TlyOQajCct35tgFJgAwW+qwfGcODEaTk2ZGRET9GUMTuRSrTWBjah5EB8daHtuYmgerraMRREREvYehiVxKVn55uzNMrQkAJksdsvLL+25SREREYGgiF1NS3Xlg6so4IiKinsLQRC4lMti/R8cRERH1FN49R33uZlUCU2PDEK3yh9lS1+G6JgUAtar5Z4iIiPoSQxP1qVtVCXh7KbA+WYvlO3OgAOyCU0tD0/pkLfuaiIioz/HyHPUZuVUCel00ti+ZDLXK/hKcWuWP7Usms6eJiIicgmeaqE/cqkpAgeYqgflaNby9FNDrojFfq2YjOBERuQyGJuoTjlQJJIwMBwB4eymk/01ERORsDE3UI261TxyrBIiIyN0xNFG3ydknjlUCRETk7rgQnLpF7uLuliqBzlYkKdActFglQEREroqhibrMkX3iWqoEALQLTqwSICIid8DQRF3m6D5xrBIgIiJ3xjVNdFM3W+DdlcXdrBIgIiJ3xdBEnbrVAu+uLu5mlQAREbkjXp6jDslZ4M3F3URE1J8wNFE7chd4A+DibiIi6jcYmqgdRxZ4c3E3ERH1F1zT1A/1dHs3F3cTEVF/wNDUz/RWezcXdxMRkafj5bl+hO3dREREXcfQ1E+wvZuIiKh7GJo8iNUmkHmuDF8cv4TMc2Ww2m5EJLZ3ExERdQ/XNHmIW61VYns3ERFR9zA0eYCWtUptL721rFXavmQy27uJiIi6iZfn3JzctUrxw0K5uJuIiKgbGJrcnNy1StkXK7i4m4iIqBsYmtycI2uVuLibiIio67imycXdqr3b0bVKXNxNRETUNQxNLkxOe3dLEaXZUtfhuiYFms8ktV6rxMXdREREjuPlORclt72bRZRERER9g6HJBTnS3g2wiJKIiKgv8PKcC3KkvbvlMhvXKhEREfUuhiYX1JX2boBrlYiIiHoTL8+5oK62dxMREVHvYWhyQS13xLG9m4iIyHUwNLkg3hFHRETkehiaXBTviCMiInItXAjuwnhHHBERketgaHJxvCOOiIjINfDyHBEREZEMDE1EREREMjA0EREREcnA0EREREQkA0MTERERkQwMTUREREQyMDS18dZbbyE2Nhb+/v6Ij4/Ht99+6+wpERERkQtgaGrlo48+wurVq/H888/j2LFjuOuuu3DvvfeioKDA2VMjIiIiJ1MIIYSzJ+Eqpk2bhsmTJ2P79u3SY+PHj8f999+PzZs3txtfX1+P+vp66fuqqirExMTAYrEgJCSkT+ZMRERE3VNVVQWVSnXLz282gl/X0NCA7OxsPPfcc3aPJyYm4tChQx3+zObNm7Fx48Z2j1dVVfXKHImIiKjntXxu3+o8EkPTdVeuXIHVakVUVJTd41FRUTCbzR3+zLp16/D0009L31+6dAlarRYxMTG9OlciIiLqedXV1VCpVJ0eZ2hqQ6Gw3wxXCNHusRZKpRJKpVL6PigoCHl5edBqtSgsLOQlOhfVchmV75Hr4nvk+vgeuT6+R/IJIVBdXQ2NRnPTcQxN10VERMDb27vdWaWSkpJ2Z5864+XlhcGDBwMAQkJC+B+pi+N75Pr4Hrk+vkeuj++RPDc7w9SCd89d5+fnh/j4eGRkZNg9npGRgRkzZjhpVkREROQqeKaplaeffhopKSmYMmUKEhIS8O6776KgoACPP/64s6dGRERETsbQ1MqDDz6IsrIyvPjiizCZTNDpdNi9ezeGDRsm+zmUSiXWr19vt9aJXAvfI9fH98j18T1yfXyPeh57moiIiIhk4JomIiIiIhkYmoiIiIhkYGgiIiIikoGhiYiIiEgGhqYuunTpEpYsWYLw8HAMGDAAt99+O7Kzs6XjQghs2LABGo0GAQEBmD17NnJzc5044/5l+PDhUCgU7b6eeOIJAHx/XEFTUxNeeOEFxMbGIiAgACNGjMCLL74Im80mjeH75HzV1dVYvXo1hg0bhoCAAMyYMQNHjx6VjvM96lvffPMNkpOTodFooFAo8Pnnn9sdl/N+1NfXY+XKlYiIiEBgYCAWLVqEoqKiPnwVbkyQw8rLy8WwYcPEww8/LI4cOSLy8/PFnj17xNmzZ6UxW7ZsEcHBweKTTz4RJ06cEA8++KCIjo4WVVVVTpx5/1FSUiJMJpP0lZGRIQCIffv2CSH4/riCl156SYSHh4v/+7//E/n5+eIf//iHCAoKEm+++aY0hu+T8z3wwANCq9WKAwcOiDNnzoj169eLkJAQUVRUJITge9TXdu/eLZ5//nnxySefCADis88+szsu5/14/PHHxeDBg0VGRobIyckRc+bMERMnThRNTU19/GrcD0NTFzz77LNi5syZnR632WxCrVaLLVu2SI/V1dUJlUol3n777b6YIrXx1FNPiZEjRwqbzcb3x0UsWLBAPPLII3aP/fSnPxVLliwRQvD/R67g2rVrwtvbW/zf//2f3eMTJ04Uzz//PN8jJ2sbmuS8H5WVlcLX11fs2rVLGnPp0iXh5eUlDAZDn83dXfHyXBd8+eWXmDJlCn72s58hMjISkyZNwnvvvScdz8/Ph9lsRmJiovSYUqnErFmzcOjQIWdMuV9raGjAzp078cgjj0ChUPD9cREzZ87EP//5T/z4448AgB9++AEHDx7EfffdB4D/P3IFTU1NsFqt8Pf3t3s8ICAABw8e5HvkYuS8H9nZ2WhsbLQbo9FooNPp+J7JwNDUBefPn8f27dsxevRopKWl4fHHH8eqVavw17/+FQCkTX/bbvQbFRXVbkNg6n2ff/45Kisr8fDDDwPg++Mqnn32WTz00EMYN24cfH19MWnSJKxevRoPPfQQAL5PriA4OBgJCQn4/e9/j8uXL8NqtWLnzp04cuQITCYT3yMXI+f9MJvN8PPzQ2hoaKdjqHPcRqULbDYbpkyZgk2bNgEAJk2ahNzcXGzfvh2/+MUvpHEKhcLu54QQ7R6j3ve///u/uPfee6HRaOwe5/vjXB999BF27tyJv/3tb4iLi8Px48exevVqaDQaLF26VBrH98m53n//fTzyyCMYPHgwvL29MXnyZCxevBg5OTnSGL5HrqUr7wffM3l4pqkLoqOjodVq7R4bP348CgoKAABqtRoA2qX2kpKSdv8CoN518eJF7NmzB7/61a+kx/j+uIbf/va3eO655/Dzn/8cEyZMQEpKCn7zm99g8+bNAPg+uYqRI0fiwIEDuHr1KgoLC5GVlYXGxkbExsbyPXIxct4PtVqNhoYGVFRUdDqGOsfQ1AV33nknTp8+bffYjz/+KG3s2/KXSUZGhnS8oaEBBw4cwIwZM/p0rv3dn//8Z0RGRmLBggXSY3x/XMO1a9fg5WX/V5C3t7dUOcD3ybUEBgYiOjoaFRUVSEtLw09+8hO+Ry5GzvsRHx8PX19fuzEmkwlGo5HvmRzOXIXurrKysoSPj494+eWXxZkzZ8QHH3wgBgwYIHbu3CmN2bJli1CpVOLTTz8VJ06cEA899BBvw+1jVqtVDB06VDz77LPtjvH9cb6lS5eKwYMHS5UDn376qYiIiBDPPPOMNIbvk/MZDAbx9ddfi/Pnz4v09HQxceJEMXXqVNHQ0CCE4HvU16qrq8WxY8fEsWPHBADx+uuvi2PHjomLFy8KIeS9H48//rgYMmSI2LNnj8jJyRH33HMPKwdkYmjqotTUVKHT6YRSqRTjxo0T7777rt1xm80m1q9fL9RqtVAqleLuu+8WJ06ccNJs+6e0tDQBQJw+fbrdMb4/zldVVSWeeuopMXToUOHv7y9GjBghnn/+eVFfXy+N4fvkfB999JEYMWKE8PPzE2q1WjzxxBOisrJSOs73qG/t27dPAGj3tXTpUiGEvPejtrZWPPnkkyIsLEwEBASIhQsXioKCAie8GvejEEIIJ57oIiIiInILXNNEREREJANDExEREZEMDE1EREREMjA0EREREcnA0EREREQkA0MTERERkQwMTUREREQyMDQRERERycDQRERERCQDQxMR9StCCMybNw9JSUntjr311ltQqVQoKChwwsyIyNUxNBFRv6JQKPDnP/8ZR44cwTvvvCM9np+fj2effRZ/+MMfMHTo0B79nY2NjT36fETkHAxNRNTvxMTE4A9/+APWrl2L/Px8CCHw6KOPYu7cuZg6dSruu+8+BAUFISoqCikpKbhy5Yr0swaDATNnzsTAgQMRHh6OhQsX4ty5c9LxCxcuQKFQ4O9//ztmz54Nf39/7Ny50xkvk4h6GDfsJaJ+6/7770dlZSX+/d//Hb///e9x9OhRTJkyBcuWLcMvfvEL1NbW4tlnn0VTUxP27t0LAPjkk0+gUCgwYcIE1NTU4He/+x0uXLiA48ePw8vLCxcuXEBsbCyGDx+O//qv/8KkSZOgVCqh0Wic/GqJqLsYmoio3yopKYFOp0NZWRk+/vhjHDt2DEeOHEFaWpo0pqioCDExMTh9+jTGjBnT7jlKS0sRGRmJEydOQKfTSaHpzTffxFNPPdWXL4eIehkvzxFRvxUZGYnHHnsM48ePx7/9278hOzsb+/btQ1BQkPQ1btw4AJAuwZ07dw6LFy/GiBEjEBISgtjYWABot3h8ypQpfftiiKjX+Th7AkREzuTj4wMfn+a/Cm02G5KTk7F169Z246KjowEAycnJiImJwXvvvQeNRgObzQadToeGhga78YGBgb0/eSLqUwxNRETXTZ48GZ988gmGDx8uBanWysrKcPLkSbzzzju46667AAAHDx7s62kSkZPw8hwR0XVPPPEEysvL8dBDDyErKwvnz59Heno6HnnkEVitVoSGhiI8PBzvvvsuzp49i7179+Lpp5929rSJqI8wNBERXafRaPDdd9/BarUiKSkJOp0OTz31FFQqFby8vODl5YVdu3YhOzsbOp0Ov/nNb/Dqq686e9pE1Ed49xwRERGRDDzTRERERCQDQxMRERGRDAxNRERERDIwNBERERHJwNBEREREJANDExEREZEMDE1EREREMjA0EREREcnA0EREREQkA0MTERERkQwMTUREREQy/H8XjPqXvWktUQAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "next year Salary in Taiwan would be: $51863.86965605492\n",
      "Salary in Taiwan at 150 would be: $69460.53971793069\n"
     ]
    }
   ],
   "source": [
    "def predict(X, weight):\n",
    "    return weight * X\n",
    "\n",
    "\n",
    "def gradient(data, Y, alpha, step):\n",
    "    m = len(data)\n",
    "    weight = 1\n",
    "    for i in range(0, step):\n",
    "        sum = 0\n",
    "        for j in range(0, m):\n",
    "            prediction = weight * data[j]\n",
    "\n",
    "            errors = prediction - Y[j]\n",
    "            sum += errors * data[j]\n",
    "        if (i % 100 == 0):\n",
    "            print(weight)\n",
    "            print(data)\n",
    "        gradient = (1/m) * alpha * errors\n",
    "        weight = weight - gradient\n",
    "\n",
    "    return weight\n",
    "\n",
    "w = gradient(train_data, train_label, 0.01, 300)\n",
    "plt.plot(train_data, w * train_data)\n",
    "plt.scatter(train_data, train_label)\n",
    "plt.xlabel(\"Year\")\n",
    "plt.ylabel(\"Salary(TWD)\")\n",
    "plt.show()\n",
    "\n",
    "print(\"next year Salary in Taiwan would be: \" + \"$\" + str(w * 112))\n",
    "print(\"Salary in Taiwan at 150 would be: \" + \"$\" + str(w * 150))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94812d79",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
